 ---
title: "Identity Protocol for APCL project "
output: html_notebook
params:
 data_dir: "seq03-33_identity"
 data_name: "33-03_seq_identity"
 ---
# Workspace setup
```{r setup}
library(tidyverse)
library(lubridate)
library(RMySQL)
library(here)
library(clownfish)
library(janitor) # clean_names
# source("../laboratory/scripts/lab_helpers.R")
# source("scripts/gen_helpers.R")
# leyte <- read_db("Leyte")
# lab <- read_db("Laboratory")
# 2019-03-06 have to add some code to get functions to work after updating packages and R
source("../../clownfish-pkg/R/db_connections.R")
leyte <- read_db("Leyte")
lab <- read_db("Laboratory")
```

This notebook continues where filtering left off, moving the genepop into cervus and processing identity results.
 
## Import the genepop to cervus
The genepop should be stored either on google drive or github and downloaded onto the windows machine.

**Because I did not run the process genepop script first, I had to open the file and add a "pop" line between the contig header and the first sample info**
 
###Open Cervus and convert the genepop to ceruvs:
 - Tools > Convert genotype file > Genepop to Cervus
 - A window will open, navigate to the genepop file that has been downloaded to the windows machine
 - Choose 2 digit format, do not use the first id as a population name
 **(2893 individuals, 1447 loci)**
 
### Run an allele frequency analysis:
 - Analysis > Allele frequency analysis
 - make sure the converted file is in the genotype file field
 * Yes Header Row
 * Yes Read Locus Names
 * ID in column 2
 * First allele in column 3
 * Type in number of loci from conversion (1447)
 * Save as “….._AF"
 * Leave all the output option boxes checked
  
### Run an identity analysis
 * Genotype file should be autofilled with the convert file created above
 * Yes Header Row
 * Id in column 2
 * First allele in column 3
 * Do not test sexes separately
 * Allele freq data should be autofilled with the AF file created above
 * Save as “…._ID"
 * Fill in minimum number of matching loci (80% of total = 1158)
 * allow fuzzy matching with 10% mismatch (10% of total = 144)
 * don’t show all comparisons - this generates a HUGE csv of every single pairwise comparison.
 
### Upload the output to github
 
### Import the data into R
 
```{r}
idcsv <- read_csv(here("data","seq03-33_identity/33-03_seq_identity_ID.csv"), col_types = cols(
  `First ID` = col_character(),
  `Loci typed` = col_integer(),
  `Second ID` = col_character(),
  `Loci typed_1` = col_integer(),
  `Matching loci` = col_integer(),
  `Mismatching loci` = col_integer(),
  pID = col_double(),
  pIDsib = col_double(),
  Status = col_character()
)) %>% 
  clean_names() %>% 
  rename(first_loci_typed = loci_typed, 
         second_loci_typed = loci_typed_1)

names(idcsv)

# add mismatch rate
idcsv <- idcsv %>%
  mutate(mismatch_prop = mismatching_loci/(mismatching_loci + matching_loci)) %>% 
  # order by number of matching loci
  arrange(matching_loci) %>% 
  # for the current run, there are extra .F on some names, remove them
mutate(first_id = ifelse(grepl(".F", first_id), substr(first_id, 1, 10), first_id), 
       second_id = ifelse(grepl(".F", second_id), substr(second_id, 1, 10), second_id)) %>% 
  # remove APCL_ from IDs
  separate(first_id, c("spp", "first_id"), sep = "_") %>% 
  separate(second_id, c("spp2", "second_id"), sep = "_") %>% 
  select(-spp, -spp2)


ggplot(idcsv, aes(x = matching_loci, y = mismatch_prop)) +
  geom_point() +
  theme_bw()
 
```

  ### Add sample_ids
```{r}
 lab <- read_db("Laboratory")
 
 # for first_sample_id
 temp <- idcsv %>% 
   rename(ligation_id = first_id)
 lab1 <- samp_from_lig(temp)
 names(lab1) <- paste("first_", names(lab1), sep = "")
 
 
 # for second_sample_id
 temp <- idcsv %>% 
   rename(ligation_id = second_id)
 lab2 <- samp_from_lig(temp)
 names(lab2) <- paste("second_", names(lab2), sep = "")
 
 # set aside all of the regenotyped samples
 regenos <- idcsv %>%
   filter(grepl("\\.", first_id) | grepl("\\.", second_id))
 
idcsv <- anti_join(idcsv, regenos, by = c("first_id", "first_loci_typed", "second_id", "second_loci_typed", "matching_loci", "mismatching_loci", "p_id", "p_i_dsib", "status", "mismatch_prop"))

 idcsv <- left_join(idcsv, lab1, by = c("first_id" = "first_ligation_id"))
 idcsv <- left_join(idcsv, lab2, by = c("second_id" = "second_ligation_id"))
 
 
# for regenos, there are potentially 6 rounds of ligations but because they are regenotypes, only have to find sample names twice

regenos <- regenos %>% 
   # keep the original names so that these can be matched back into the main data set
   mutate(keep1 = first_id, 
          keep2 = second_id) %>% 
   # make sure to double escape the . 
   separate(first_id, c("ligation_id", "half2"), sep = "\\.") %>% 
   separate(second_id, c("half3", "half4"), sep = "\\.")
 
 samp1 <- samp_from_lig(regenos)
 regenos <- left_join(regenos, samp1, by = "ligation_id") %>% 
   # change column names to find sample ids for next round
   rename(half1 = ligation_id, 
          ligation_id = half3) 
 regenos <- regenos %>% 
   rename(first_sample_id = sample_id)
 samp2 <- samp_from_lig(regenos)
 regenos <- left_join(regenos, samp2, by = "ligation_id") %>% 
   # remove extra columns and change names
   select(-contains("half")) %>% 
   rename(first_id = keep1, 
          second_id = keep2) 
 
 regenos <- regenos %>% 
   select(-ligation_id) %>% 
   rename(second_sample_id = sample_id)
 
 # rejoin regenos to the idcsv
 
 idcsv <- rbind(idcsv, regenos)
 rm(regenos, samp1, samp2, lab1, lab2, temp)
```
  
# ### Add field data
```{r}
 first <- fish_anem_dive() %>% 
  filter(sample_id %in% idcsv$first_sample_id)
 names(first) <- paste0("first_", names(first))
 sec <- fish_anem_dive() %>% 
   filter(sample_id %in% idcsv$second_sample_id)
names(sec) <- paste0("second_", names(sec))


idcsv <- left_join(idcsv, first, by = "first_sample_id")
idcsv <- left_join(idcsv, sec, by = "second_sample_id")

# rearrange columns
idcsv <- idcsv %>% 
  select(contains("first"), contains("second"), everything())

rm(first, sec)

```

## Get lat lons
```{r}

#find the lat lon of the first anem
first <- idcsv %>% 
  mutate(first_anem_obs_time = force_tz(ymd_hms(str_c(first_date, first_anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(first_anem_obs_time = with_tz(first_anem_obs_time, tzone = "UTC")) %>% 
  mutate(first_hour = hour(first_anem_obs_time), 
         first_minute = minute(first_anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% idcsv$first_date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
first <- left_join(first, lat, by = c("first_date" = "gpx_date", "first_hour" = "gpx_hour", "first_minute" = "minute", "first_gps" = "unit"))

# summarize the lat lons
first <- first %>% 
  group_by(first_id) %>% 
  summarise(first_lat = mean(as.numeric(lat)), 
            first_lon = mean(as.numeric(lon)))

idcsv <- left_join(idcsv, first, by = "first_id")
rm(first, lat)

# find the lat lon of the second anem
second <- idcsv %>% 
  mutate(second_anem_obs_time = force_tz(ymd_hms(str_c(second_date, second_anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(second_anem_obs_time = with_tz(second_anem_obs_time, tzone = "UTC")) %>% 
  mutate(second_hour = hour(second_anem_obs_time), 
         second_minute = minute(second_anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% idcsv$second_date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
second <- left_join(second, lat, by = c("second_date" = "gpx_date", "second_hour" = "gpx_hour", "second_minute" = "minute", "second_gps" = "unit"))

# summarize the lat lons
second <- second %>% 
  group_by(second_id) %>% 
  summarise(second_lat = mean(as.numeric(lat)), 
            second_lon = mean(as.numeric(lon)))

idcsv <- left_join(idcsv, second, by = "second_id")
rm(lat, second)  

```
### Pull out new regenotypes
```{r}
new_regenos <- idcsv %>% 
  filter(first_sample_id == second_sample_id)

idcsv <- anti_join(idcsv, new_regenos)
```


### Flag fish that were caught on the same day - the only one is a fish that has already been flagged as a recapture
```{r}
idcsv <- idcsv %>% 
  mutate(date_eval = ifelse(first_date == second_date, "FAIL", NA))

date_fails <- idcsv %>% 
  filter(date_eval == "FAIL") %>% 
  select(first_sample_id, second_sample_id, first_gen_id, second_gen_id, everything())

```

### Flag matches that were caught more than 250m apart.  Given the data that Katrina has found as well as my data, it looks like it is possible that these fish have moved sites and should not be discounted simply because they weren't captured at the same site.  These need to be looked into.

```{r}
# calculate the distances
alldists <- fields::rdist.earth(as.matrix(idcsv[,c("first_lon", "first_lat")]), as.matrix(idcsv[,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

idcsv <- idcsv %>% 
  mutate(distkm = diag(alldists), 
  dist_eval = ifelse(distkm > 0.25, "FAIL", NA)
         )

dist_fails <- idcsv %>% 
  filter(dist_eval == "FAIL") %>% 
  select(first_site, second_site, first_gen_id, second_gen_id, first_tag_id, second_tag_id, first_sample_id, second_sample_id, everything())

# how many are at the same site? 2
same_site <- dist_fails %>% 
  filter(first_site == second_site)

# test <- dist_fails %>% 
  # filter(first_sample_id == "APCL15_098")

```
### Flag any fish that shrink by more than 1.5cm  
- one pair is clearly not truely the same fish (L4964, L5012) - caught at Wangag and Tamakin days from each other with different pit tags

```{r}
idcsv <- idcsv %>% 
  mutate(size_eval = ifelse(first_date < second_date & as.numeric(first_size) > as.numeric(second_size) + 1.5, "FAIL", NA), 
         size_eval = ifelse(first_date > second_date & as.numeric(first_size) + 1.5 < as.numeric(second_size), "FAIL", NA))

size_fails <- idcsv %>% 
  filter(size_eval == "FAIL") %>% 
  select(first_id, second_id, first_date, second_date, first_size, second_size, first_gen_id, second_gen_id, first_tag_id, second_tag_id, mismatch_prop, everything())
```

### Flag fish that were caught at different sites
```{r}
idcsv <- idcsv %>% 
  mutate(site_eval = ifelse(first_site != second_site, "FAIL", NA))

site_fails <- idcsv %>% 
  filter(site_eval == "FAIL") %>% 
  select(first_site, second_site, first_gen_id, second_gen_id, first_tag_id, second_tag_id, first_id, second_id, everything())
```


# Write big table into a file -----------------------
```{r}
# #write_csv(idcsv, paste0("data/", params$data_dir, "/", params$data_name, "_big_table.csv"))
# #write_csv(new_regenos, paste0("data/", params$data_dir, "/new_regenos.csv"))

# idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")

# cleanup
rm(alldists)
```
# Does the first gen_id match the second gen id?
```{r}
old_pairs <- idcsv %>% 
  filter(first_gen_id == second_gen_id, 
         !is.na(first_gen_id)) %>% 
  select(first_gen_id, second_gen_id, everything())
```


# Examine each new match
```{r}
new_pairs <- anti_join(idcsv, old_pairs)

i <- 1
  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second)
  

```
Really have to go through all of these by hand and make sure they look good.

I'm pulling the matches into a table and looking at the data side by side, especially size, color, tag_id, and lat lon.  It is interesting how many of these matches are from fish that were caught a few years apart at almost exactly the same lat lon.

First pair, APCL15_375493 & APCL16_040, look great.  Adding the gen_id of the 2015 fish to the 2016 fish. Tag loss
```{r}
# leyte <- read_db("Leyte")

fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

fish <- fish %>%
  mutate(gen_id = ifelse(sample_id == "APCL16_040", 1483, gen_id))
```
Next set, look good, not the same anemone but the same site and similar coordinates, looks like the fish grew from an Orange to a YP. APCL15_405714 & APCL17_048. Tag loss.
```{r}
fish <- fish %>%
mutate(gen_id = ifelse(sample_id == "APCL17_048", 1603, gen_id))
```
Next set look good, fish grew, tag loss, same site, APCL15_369033 & APCL16_630
```{r}
fish <- fish %>%
  mutate(gen_id = ifelse(sample_id == "APCL16_630", 1369, gen_id))
```
This next set needs to be investigated. 2 fish are from the same year, same site, 1cm difference in size, the third in the set is from 2014 and a different site but could have moved.
```{r}
i <- 4

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct()
  
  investigate <- tibble()
  
  z <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  investigate <- rbind(investigate, z)
```

i <- 5, Same as the one above because the first_id is repeated for 2 different matches


next bunch
```{r}
i <- 6

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id == y$sample_id[2], y$gen_id[1], gen_id))
```

next bunch
```{r}
i <- 7

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
    mutate(gen_id = ifelse(sample_id == y$sample_id[2], y$gen_id[1], gen_id))
```


next bunch - tagged recapture and took a finclip anyway and it matches, very nice
```{r}
i <- 8

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
    mutate(gen_id = ifelse(sample_id == y$sample_id[2], y$gen_id[1], gen_id))
```


next bunch 
```{r}
i <- 9

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
    mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1465, gen_id))
```
10 - same as the ones above
next bunch  
```{r}
i <- 11

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
    mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1378, gen_id))
```

next bunch  
```{r}
i <- 12

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1433, gen_id))
```


next bunch  
```{r}
i <- 13

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 999, gen_id))
```

next bunch  
```{r}
i <- 14

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1040, gen_id))
```


next bunch  - these fish need to be investigated - different sites, about 1km away, same size but color change from YP to O.
```{r}
i <- 15

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  
   investigate <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1040, gen_id))
```

size, color, location all look great
```{r}
i <- 16

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1372, gen_id))
```

```{r}
i <- 17

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  
  # This matches to an old id error
  z <- c("APCL14_305", "APCL14_306")
  test <- fish_anem_dive(z, leyte)
  # even though there is a match, can't tell which fish it was because they were caught on the same anemone in the same year
  
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1372, gen_id))
```

need investigation because 2 fish are from the same year, different sites
```{r}
i <- 18

  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6)) %>% 
    arrange(year)
  fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1372, gen_id))
   
  z <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  investigate <- rbind(investigate, z)
  
```

# need investigation because 2 fish are from the same year, different sites, i=20 is same set
# ```{r}
# i <- 19
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6)) %>% 
#     arrange(year)
#   fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1372, gen_id))
#   
#   z <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   investigate <- rbind(investigate, z)
# ```
# 
# need investigation because 2 fish are from the same year, different sites
# ```{r}
# i <- 21
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6)) %>% 
#     arrange(year)
#   fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1372, gen_id))
#   
#   z <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   investigate <- rbind(investigate, z)
# ```
# 
# Even though this fish goes from a YP to an O, it wasn't the largest fish on the anemone and so was not the YP breeder on the anemone when it was marked as YP.
# ```{r}
# i <- 22
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6)) %>% 
#     arrange(year)
#   
#   fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1557, gen_id))
# ```
# 
# 
# needs investigation, didn't grow enough, different sites
# ```{r}
# i <- 23
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6)) %>% 
#     arrange(year)
#   fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1557, gen_id))
# 
#   z <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   investigate <- rbind(investigate, z) %>% 
#     distinct()
# ```
# 
# 
# ```{r}
# i <- 24
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6), 
#            size = as.numeric(size)) %>% 
#     arrange(year)
#   
#   
#   if (!is.unsorted(y$size)){
#     if (nrow(distinct(y,site)) == 1){
#       new_pairs$distkm[i]
#       temp <- y %>% 
#         filter(!is.na(gen_id)) %>% 
#         distinct(gen_id)
#       if (nrow(temp) == 1){
#   (fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id)))
#       }else{
#         print("More than one gen_id!")
#           (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#       }
#     }else{
#       print("Fish moved sites")
#          (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
# (  investigate <- rbind(investigate, z) %>%
#     distinct())
#     }
#   }else{
#     print("Fish shrinks!")
#        (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#   }
#     
#     
#   
#  
#   
# 
# ```
# ```{r}
# i <- 25
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6), 
#            size = as.numeric(size)) %>% 
#     arrange(year)
#   
#   if (length(unique(y$year)) == nrow(y)){
#   if (!is.unsorted(y$size)){
#     if (nrow(distinct(y,site)) == 1){
#       new_pairs$distkm[i]
#       temp <- y %>% 
#         filter(!is.na(gen_id)) %>% 
#         distinct(gen_id)
#       if (nrow(temp) == 1){
#   (fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id)))
#       }else{
#         print("More than one gen_id!")
#           (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#       }
#     }else{
#       print("Fish moved sites")
#          (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
# (  investigate <- rbind(investigate, z) %>%
#     distinct())
#     }
#   }else{
#     print("Fish shrinks!")
#        (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#   }
#   }else{
#     print("Fish caught in same year!")
#        (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#   }
# ```
# 
# ```{r}
# i <- 26
# 
#   x <- new_pairs %>% 
#     filter(first_id == new_pairs$first_id[i])
#   first <- x %>% 
#     select(contains("first"))
#   second <- x %>% 
#     select(contains("second"))
#   names(first) <- substr(names(first), 7, 20)
#   names(second) <- substr(names(second), 8, 25)
#   y <- rbind(first, second) %>% 
#     distinct() %>% 
#     mutate(year = substr(sample_id, 5,6), 
#            size = as.numeric(size)) %>% 
#     arrange(year)
#   
#   if (length(unique(y$year)) == nrow(y)){
#   if (!is.unsorted(y$size)){
#     if (nrow(distinct(y,site)) == 1){
#       new_pairs$distkm[i]
#       temp <- y %>% 
#         filter(!is.na(gen_id)) %>% 
#         distinct(gen_id)
#       if (nrow(temp) == 1){
#   (fish <- fish %>%
#   mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id)))
#       }else{
#         print("More than one gen_id!")
#           (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#       }
#     }else{
#       print("Fish moved sites")
#          (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
# (  investigate <- rbind(investigate, z) %>%
#     distinct())
#     }
#   }else{
#     print("Fish shrinks!")
#        (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#   }
#   }else{
#     print("Fish caught in same year!")
#        (z <- new_pairs %>%
#     filter(first_id == new_pairs$first_id[i]))
#   (investigate <- rbind(investigate, z) %>%
#     distinct())
#   }
# ```
# Assign new pairs automatically #######################################
```{r}
investigate <- tibble()
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

for(i in seq(new_pairs$first_id)){
  print(i)
  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  temp <- y %>% 
    filter(is.na(fish_table_id))
  if (nrow(temp) == 0){
    if (length(unique(y$year)) == nrow(y)){
      if (!is.unsorted(y$size)){
        if (nrow(distinct(y,site)) == 1){
          temp <- y %>% 
            filter(!is.na(gen_id)) %>% 
            distinct(gen_id)
          if (nrow(temp) == 1){
            fish <- fish %>%
               mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id))
          
           }else{
              temp <- y %>% 
              distinct(gen_id)
            if (nrow(temp) == 1 & is.na(temp$gen_id)){
              y <- create_genid(y)
              fish <- fish %>%
               mutate(gen_id = ifelse(sample_id %in% y$sample_id, y$gen_id[1], gen_id))
            }else{
                z <- new_pairs %>%
              filter(first_id == new_pairs$first_id[i]) %>% 
              mutate(id_note = "More than one gen_id!")
            investigate <- rbind(investigate, z) %>%
              distinct()
              }
            
            }
        }else{
          z <- new_pairs %>%
            filter(first_id == new_pairs$first_id[i]) %>% 
            mutate(id_note = "Fish moved sites")
          investigate <- rbind(investigate, z) %>%
            distinct()
        }
      }else{
        z <- new_pairs %>%
          filter(first_id == new_pairs$first_id[i]) %>% 
          mutate(id_note = "Fish shrinks!")
        investigate <- rbind(investigate, z) %>%
          distinct()
      }
    }else{
      z <- new_pairs %>%
        filter(first_id == new_pairs$first_id[i]) %>% 
        mutate(id_note = "Fish caught in same year!")
      investigate <- rbind(investigate, z) %>%
        distinct()
    }
  }else{
    z <- new_pairs %>%
      filter(first_id == new_pairs$first_id[i]) %>% 
      mutate(id_note = "Fish had previous identity issues!")
    investigate <- rbind(investigate, z) %>%
      distinct()
  }
}

investigate <- investigate %>% 
  mutate(first_tag_id = as.character(first_tag_id), 
         second_tag_id = as.character(second_tag_id))
```
# 
# # Check the fish that have been assigned as genetic recaptures automatically
# ```{r}
# test <- fish %>% 
#   group_by(gen_id) %>% 
#   summarise(count = n()) %>% 
#   filter(count > 1)
# 
# test1 <- fish %>% 
#   filter(gen_id %in% test$gen_id, 
#          !is.na(gen_id)) %>% 
#   arrange(gen_id)
# ```
# 
# Send the fish that appear to be good recaptures based on size and distance to the database
```{r}
# backup the database
clown <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

#write_csv(clown, path = paste0("../db_backups/", Sys.Date(), "_clownfish_db.csv"))

# can send the fish table to the database because I've been updating it all along
# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, row.names = F, overwrite = T)
# dbDisconnect(ley)

rm(fish, clown)
```
## Go through the investigate table
# i1
```{r}
i <-1

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
For a fish that changed sites, I can look at the lat lon of all of the fish in the match and create a csv, then pop that into q-gis and make note of any other fish that were captured that year in that space (if I caught a fish in 2014 at Magbangon and caugth the same genetic fish at Palanas in 2016, which fish from the palanas coords did I catch in 2014 and which fish from the Magbangon coords did I catch in 2016 and then look at if it is possible for any of those candidates to be a match (were they smaller/bigger) and look at the labwork and see if there was any possiblity for mix up).  Instead of QGIS I could just look at distance between captures but QGIS might be quicker.

```{r compare fish from sites}
# For a fish caputred in 2014 at Magbangon that matches 2 fish captured in Palanas in 2016
source("../clownfish-pkg/R/db_connections.R")
leyte <- read_db("Leyte")

# find the 2014 fish in Palanas
# get dives
pal14 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% pal14$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

pal14 <- left_join(temp, pal14, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% pal14$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

pal14 <- left_join(temp, pal14, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- pal14 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% pal14$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

pal14 <- left_join(pal14, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(pal14[,c("lon", "lat")]), as.matrix(investigate[1,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

pal14 <- pal14 %>% 
  mutate(distkm = alldists) 

# fish that were caught within 20 m of the 2016 fish
pal14_small <- pal14 %>% 
# the 2014 fish has to be smaller than 8.3cm
filter(size < 8.3)

# out of these 7 fish, one was unsuccessfully genotyped, could it be the one that was mixed up?  APCL14_553


pal14_work <- work_history(pal14_small, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- pal14_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- pal14_work %>% 
  filter(plate.y %in% y_work$plate.y)
same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"

same_lig <- pal14_work %>% 
  filter(plate %in% y_work$plate)

same_barcode <- pal14_work %>% 
  filter(barcode_num %in% y_work$barcode_num)

same_pool <- pal14_work %>% 
  filter(pool %in% y_work$pool)
```
During the ligation, I was supposed to pull APCL14_454 from well G10 into D11 and it is possible I pulled APCL14_552 from well D10 instead.  However, if that were the case L0626 and L0624 would match each other, and they don't.  

Out of all pal14 that are smaller than the fish in 2016, 3 have the same barcode as the 2014 fish.
```{r}
one <- pal14 %>% 
  filter(sample_id %in% c("APCL14_174", "APCL14_158", "APCL14_089")) %>% 
  select(sample_id, size, color, lat, lon, gen_id)

two <- y %>% 
  select(sample_id, size, color, lat, lon, gen_id) 

temp <- rbind(one, two)
```
All of these have gen_id's, meaning the DNA juice was successfully genotyped, so even if there is an incorrect label on the ligation that is producing this match, it does not match to any of the above labels.  The only conclusion we can come to from this analysis is that it is possible for another sample to have been mixed up with this one.  If another sample was pulled twice, once into the location it was supposed to go and once into this location, it would be a match.  If a sample was only pulled once and put into the incorrect location, the original label would be ungenotyped.
```{r}
# how many of pal14 that are small enough to match our fish are ungenotyped? 8
non <- pal14_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id)
temp <- rbind(two, non)
  
```

# i1 decision 
Move the 2014 sample to the known_issues group, name the 2016 samples as matches and assign gen_id
```{r}
# max_gen <- fish %>% 
#   summarise(max = max(gen_id, na.rm = T)) %>% 
#   collect()
# 
# fish <- fish %>% 
#   mutate(gen_id = ifelse(sample_id == "APCL14_454", NA, gen_id), 
#          gen_id = ifelse(sample_id %in% c("APCL16_085", "APCL16_781"), max_gen$max + 1, gen_id))
```

i = 2 is the same set as above because it was a match of 3

#i3
```{r}
i <- 3

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
This fish shrank, but only by 0.4cm.  Send to database as a match
```{r}
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

temp <- y %>% 
  select(gen_id) %>% 
  filter(!is.na(gen_id)) %>% 
  distinct()

fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id))

```


#i4
```{r}
i <- 4

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
These fish are the same exact size even though 2 years have passed (suspicious), and they are at different sites.

# 2014 site 
```{r}
# For a fish caputred in 2014 at S. Magbangon that matches a fish captured in Wangag in 2016

# find the 2014 fish in Palanas
# get dives
site14 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Wangag", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% site14$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

site14 <- left_join(temp, site14, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% site14$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

site14 <- left_join(temp, site14, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- site14 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% site14$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

site14 <- left_join(site14, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(site14[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

site14 <- site14 %>% 
  mutate(distkm = alldists) 

site14_small <- site14 %>% 
# the 2014 fish has to be smaller than 8.3cm
filter(size < 9.8)
# still a ton of potential fish

# how many of site14 that are small enough to match our fish are ungenotyped? 
non <- site14_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 22 candidate fish - possibility
```

# 2016 site
```{r}
# For a fish caputred in 2014 at S. Magbangon that matches a fish captured in Wangag in 2016

# find the 2016 fish in S. Magbangon
# get dives
site16 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "S. Magbangon", 
         year(date) == "2016") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% site16$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

site16 <- left_join(temp, site16, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% site16$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

site16 <- left_join(temp, site16, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- site16 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% site16$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

site16 <- left_join(site16, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(site16[,c("lon", "lat")]), as.matrix(investigate[i,c("first_lon", "first_lat")]), miles=FALSE, R=6371) 

site16 <- site16 %>% 
  mutate(distkm = alldists) 

site16_small <- site16 %>% 
filter(size < 9.8)
# still a ton of potential fish

# how many of site16 that are small enough to match our fish are ungenotyped? 
non <- site16_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 60 candidate fish - possibility
```
 
# labwork on 2014 fish
```{r}
site14_work <- work_history(site14_small, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- site14_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- site14_work %>% 
  filter(plate.y %in% y_work$plate.y)
# same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"

same_lig <- site14_work %>% 
  filter(plate %in% y_work$plate)

same_barcode <- site14_work %>% 
  filter(barcode_num %in% y_work$barcode_num)

same_pool <- site14_work %>% 
  filter(pool %in% y_work$pool)

```


# labwork on 2016 fish
```{r}
site16_work <- work_history(site16_small, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- site16_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- site16_work %>% 
  filter(plate.y %in% y_work$plate.y)
# same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"

same_lig <- site16_work %>% 
  filter(plate %in% y_work$plate)

same_barcode <- site16_work %>% 
  filter(barcode_num %in% y_work$barcode_num)

same_pool <- site16_work %>% 
  filter(pool %in% y_work$pool)

```
# i4 decision
Move the 2014 fish to the known issues group, the 2016 fish appears fine. 
```{r}
source("../clownfish-pkg/R/db_connections.R")
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL14_544", NA, gen_id))

lb <- write_db("Laboratory")
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L0621", issue = "This sample from 2014, APCL14_544 matches L5019, APCL16_254 but they are from different sites and many fish from the correct site are on the same plate for much of the lab work leading up to this ligation.  No robots were used in the making of this ligation.")

# #dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```

#i5 
contains a match to the id error, updated note in known_issues, still can't determine which fish was the original fish
```{r}
i <- 5

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```

# i6
The 6th set contains 3 fish, one from 2015 and 2 from 2016.  The 2 from 2016 were captured a couple days apart and are from different sites.  They cannot be the same fish. 

i=7 is the same group

i=8 is the identity match from 2014 that was matched to the 2015 sample in the past.  All have the same gen_id.

i=9 is the same group.

i=10 is the 2016 fish

```{r}
i <- 6

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# labwork
Take a look at the 2016 labwork and see if they overlap at all
```{r}
y_work <- work_history(y, "sample_id") %>% 
  distinct()
```
For the two 2016 samples, they have the same barcode number (43) in pools 92 and 93.

I think the 2 samples from the same site are a match and the other has a lab issue
#i6 decision
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL16_261", 1086, gen_id))

lb <- write_db("Laboratory")
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L4978", issue = "This ligation matches L5026 and has the same barcode number in an adjacent pool but not on the same plate.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```
#i11 - different sites
Fish from different sites
```{r}
i <- 11

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# 2013 site 
```{r}
# For a fish caputred in 2013 at Visca that matches a fish captured in Poroc San Flower in 2016

# find the 2013 fish in Poroc San Flower
# get dives
site13 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Poroc San Flower", 
         year(date) == "2013") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% site13$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

site13 <- left_join(temp, site13, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% site13$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

site13 <- left_join(temp, site13, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- site13 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% site13$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

site13 <- left_join(site13, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(site13[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

site13 <- site13 %>% 
  mutate(distkm = alldists) 

site13_small <- site13 %>% 
filter(size < 9.8)
# 15 possibilites

# how many of site13 that are small enough to match our fish are ungenotyped? 
non <- site13_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 3 candidate fish
```

# 2016 site
```{r}
# # For a fish caputred in 2013 at Visca that matches a fish captured in Poroc San Flower in 2016

# find the 2016 fish in Visca
# get dives
site16 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Visca", 
         year(date) == "2016") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% site16$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

site16 <- left_join(temp, site16, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% site16$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

site16 <- left_join(temp, site16, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- site16 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% site16$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

site16 <- left_join(site16, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(site16[,c("lon", "lat")]), as.matrix(investigate[i,c("first_lon", "first_lat")]), miles=FALSE, R=6371) 

site16 <- site16 %>% 
  mutate(distkm = alldists) 

site16_small <- site16 %>% 
filter(size < 9.8)
# 9 potential fish

# how many of site16 that are small enough to match our fish are ungenotyped? 
non16 <- site16_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 5 candidate fish 
```

# labwork on 2013 fish
```{r}
site13_work <- work_history(site13_small, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- site13_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- site13_work %>% 
  filter(plate.y %in% y_work$plate.y)
# same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"

same_lig <- site13_work %>% 
  filter(plate %in% y_work$plate)

same_barcode <- site13_work %>% 
  filter(barcode_num %in% y_work$barcode_num)

same_pool <- site13_work %>% 
  filter(pool %in% y_work$pool)

```
No possible mixup of 2013 samples
# labwork on 2016 fish
```{r}
# site16_work <- work_history(non16, "sample_id")
# 
# y_work <- work_history(y, "sample_id")
# 
# same_ext <- site16_work %>% 
#   filter(plate.x %in% y_work$plate.x)
# 
# same_dig <- site16_work %>% 
#   filter(plate.y %in% y_work$plate.y)
# # same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"
# 
# same_lig <- site16_work %>% 
#   filter(plate %in% y_work$plate)
# 
# same_barcode <- site16_work %>% 
#   filter(barcode_num %in% y_work$barcode_num)
# 
# same_pool <- site16_work %>% 
#   filter(pool %in% y_work$pool)

```
all 5 of the possible 2016 fish from Visca were on the same plate as the 2016 fish. Placing the 2016 fish in the known_issues group
#i11 decision
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L5065", issue = "This ligation matches L2232 but is at a different site.  There are 5 possible fish that were caught at the right site that were also on the same plate as this ligation.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```

#i12 - shrinkage
Fish caught in the same year at the same site, very close lat lon, shrinkage of 0.9cm.
```{r}
i <- 12

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# i12 decision
These fish are a true recapture
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL16_745", 1651, gen_id))
```

#i13 - same year
These fish are from the same year, distant sites, check labwork first
```{r}
i <- 13

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```

Take a look at the 2016 labwork and see if they overlap at all
```{r}
y_work <- work_history(y, "sample_id") %>% 
  distinct()
```

#i13 decision
Once again, same barcode, pools 92 and 93, with the other samples, pool 92 was the one with the issue, adding that to issues.
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L4964", issue = "This ligation matches L5012 but that was caught in the same year at a different site.  These have the same barcode number and are in pools 92 and 93")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```

#i14 - shrinkage
```{r}
i <- 14

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
#i14 decision
This was flagged because of 0.8cm shrinkage, they are a match
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL16_788", 1508, gen_id))
  
```


#i15 - multiple gen ids
```{r}
i <- 15

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# i15 decision
These fish were flagged because they both have gen_ids and were not a match in the past.  However, the sizes and locations align for a recapture, changing gen_id of one to match the other
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL15_374990", 888, gen_id))
```



#i16 - multiple gen_ids
```{r}
i <- 16

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# i16 decision
These fish were flagged because they both have gen_ids and were not a match in the past.  However, the sizes and locations align for a recapture, changing gen_id of one to match the other
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL14_060", 654, gen_id))
```


#i17 - same year
These fish are from the same year, different sizes, different sites
```{r}
i <- 17

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```
# look at labwork among the matches
```{r eval=FALSE}
y_work <- work_history(y, "sample_id") %>% 
  distinct
```

#i17 decision
These samples were on the same extract plate and same ligation plate (near each other) and may have somehow contaiminated each other.  Move to known issues
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = c("L2572", "L2579"), issue = c("This ligation matches L2579 but is at a different site, different size, same year.  These samples could have come into contact with each other during extraction or ligation.", "This ligation matches L2572 but is at a different site, different size, same year.  These samples could have come into contact with each other during extraction or ligation."))

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```

# i 18 - shrinkage
```{r}
i <- 18

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]

```

#i18 decision
This was flagged because of 0.4cm shrinkage, they are a match
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL16_772", 1458, gen_id))
  
```


# i 19 - shrinkage of 0.6cm
i=20 is same group
```{r}
i <- 19

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]
```
#i19 decision
they are a match (all 3), even though 2 were caught in the same year, they were a month apart and in the same location
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1015, gen_id))
  
```

# i 21 - shrinkage
i=22 is same group
```{r}
i <- 21

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]
```
# i 21 decision
they are a match (all 3), even though 2 were caught in the same year, they were a month apart and in the same location, shrinkage of 0.9cm
```{r}
# double check
fish %>% filter(sample_id %in% y$sample_id)

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 755, gen_id))
  
```

#i23 - different sites

```{r}
i <- 23

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  investigate$mismatch_prop[i]
```

# 2014 site 
```{r}
# For a fish caputred in 2014 at Wangag that matches a fish captured at Palanas in 2015

# find the 2014 fish in Palanas
# get dives
site14 <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% site14$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

site14 <- left_join(temp, site14, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% site14$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

site14 <- left_join(temp, site14, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- site14 %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% site14$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

site14 <- left_join(site14, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(site14[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

site14 <- site14 %>% 
  mutate(distkm = alldists) 

site14_small <- site14 %>% 
filter(size < 8.6)
# 81 possibilites

# how many of site14 that are small enough to match our fish are ungenotyped? 
non <- site14_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 9 candidate fish
```

# 2015 site
```{r}
# get dives
siteb <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Wangag", 
         year(date) == "2015") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% siteb$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

siteb <- left_join(temp, siteb, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% siteb$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

siteb <- left_join(temp, siteb, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- siteb %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% siteb$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

siteb <- left_join(siteb, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(siteb[,c("lon", "lat")]), as.matrix(investigate[i,c("first_lon", "first_lat")]), miles=FALSE, R=6371) 

siteb <- siteb %>% 
  mutate(distkm = alldists) 

siteb_small <- siteb %>% 
filter(size < 8.6)
# 94 potential fish

# how many of siteb that are small enough to match our fish are ungenotyped? 
nonb <- siteb_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 39 candidate fish 
```

# labwork on 2014 fish
```{r eval=FALSE}
sitea_work <- work_history(non, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
# same_dig <- rbind(same_dig, y_work) # this catches all of the future "sames"

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
```
No possible mixups in the 2014 labwork

# labwork on 2015 fish
```{r eval = FALSE}
siteb_work <- work_history(nonb, "sample_id")

# only consider samples that have been ligated
# siteb_work <- siteb_work %>% 
#   filter(!is.na(ligation_id))

# out of the 39 candidates, only 4 were ligated

y_work <- work_history(y, "sample_id")

same_ext <- siteb_work %>% 
  filter(plate.x %in% y_work$plate.x, 
         !is.na(plate.x))

same_dig <- siteb_work %>% 
  filter(plate.y %in% y_work$plate.y, 
         !is.na(plate.y))


same_lig <- siteb_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(ligation_id))

same_barcode <- siteb_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- siteb_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

```
No possible mix-ups in the 2015 lab work


#i23 decision
Unlikely that there is a lab mixup because none of the fish who are potential candiates for a mistake were on the same plate as the sequencing match and the fish was captured at Wangag about 400m from where it was captured at Palanas
```{r}
fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, "recapture at different site, APCL14_408 to APCL15_376136", fish_notes))
```


#i24 - different sites
i=25 same group

This is a match of 3.  The fish from 2014 was at Wangag and the 2016 and 2017 fish were at Palanas.  The fish from Palanas are from almost the exact same spot and are a match.  Need to determine if the 2014 fish is also a match.
```{r}
i <- 24

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(size < 9.8)
# 81 possibilites

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)
# still 11 candidate fish

rm(alldists)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 same_dig <- rbind(same_dig, y_work) %>% 
   arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
rm(sitea_work, y_work, nona, nonb, sitea, sitea_small, siteb, siteb_small, siteb_work)
```
There are one or two digests that could have been mixed up. The potential for a lab error exists.

# i 24 decision
The 2 fish from the same site are recaptures but lab error cannot be ruled out for the fish from Wangag.
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L1020", issue = "This ligation matches L5203 and L3608 but was captured at Wangag and they were captured at Palanas. The possibility of lab error cannot be ruled out because fish from Palanas were on the same digest plate.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)

max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL14_422", NA, gen_id), 
         gen_id = ifelse(sample_id %in% c("APCL16_089", "APCL17_075"), max_gen$max + 1, gen_id))

# test
fish %>%
  filter(gen_id == max_gen$max + 1)

rm(iss, max_gen)
```

#i 26 - regenotype
i27 same group
i28 same group
```{r}
i <- 26

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
```
# i 26 decision
```{r}
# double check
fish %>% filter(sample_id %in% y$sample_id)

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1534, gen_id))
  
```
#i 29 - same year, same site
i30 same group
```{r}
i <- 29 

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
```
# i 29 decision - already been processed
```{r}
# double check
fish %>% filter(sample_id %in% y$sample_id)
```

#i 31 - shrinkage
```{r}
i <- 31

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
```
# i 31 decision
Shrinkage of 0.5cm, same site, same lat lon
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL17_486", 945, gen_id))
```


#i 32 - same year, same site
```{r}
i <- 32

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
```
# i 32 decision
recapture
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL13_640", 403, gen_id))
```

#i 33 - set of 3, different sites
This looks like 2 are a match and one is from a different site that may or may not be a match.
i=34 same group
```{r}
i <- 33

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
siteb <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Wangag ", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% siteb$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

siteb <- left_join(temp, siteb, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% siteb$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

siteb <- left_join(temp, siteb, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- siteb %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% siteb$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

siteb <- left_join(siteb, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(siteb[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

siteb <- siteb %>% 
  mutate(distkm = alldists) 

siteb_small <- siteb %>% 
filter(size < 9.2)

# how many of siteb that are small enough to match our fish are ungenotyped? 
nona <- siteb_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

rm(alldists, siteb, siteb_small)
```

# site B labwork
```{r eval=FALSE}
siteb_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- siteb_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- siteb_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% 
   # arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- siteb_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- siteb_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- siteb_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
rm(siteb_work, y_work, nona)
```
There are no overlaps for lab work
# decision 33 - match across different sites
```{r}
fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, "recapture at different site, APCL14_547 to APCL15_404287 and to APCL16_800", fish_notes), 
         gen_id = ifelse(sample_id == "APCL16_800", 1571, gen_id))
```


#i 35 - set of 3, regenotype
This looks like 2 are a regenotype and one is a recapture.
i = 36 same group
```{r}
i <- 35

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 35
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1231, gen_id))
```

#i 37 - shrinkage

```{r}
i <- 37

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# i 37 decision
Shrinkage of 0.5cm, same site, same lat lon
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL16_236", 1071, gen_id))
```


#i 38 - different sites

```{r}
i <- 38

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Elementary School ", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.5)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% 
   # arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
No labwork overlaps for 2014

# site B
```{r}
# get dives
siteb <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Elementary School ", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% siteb$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

siteb <- left_join(temp, siteb, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% siteb$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

siteb <- left_join(temp, siteb, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- siteb %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% siteb$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

siteb <- left_join(siteb, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(siteb[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

siteb <- siteb %>% 
  mutate(distkm = alldists) 

siteb_small <- siteb %>% 
filter(as.numeric(size) < 10.5)

# how many of siteb that are small enough to match our fish are ungenotyped? 
nonb <- siteb_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

rm(alldists, siteb, siteb_small)
```

# site B labwork
```{r eval=FALSE}
siteb_work <- work_history(nonb, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- siteb_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- siteb_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% 
   # arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- siteb_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- siteb_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- siteb_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(siteb_work, y_work, nonb)
```
No labwork overlap in 2016

```{r}
fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, "recapture at different site, APCL14_009 to APCL16_286", fish_notes))
```

#i 39 - different sites
2 of the fish are tagged and genetic recaptures, the other is at a different site
i=40 same group

```{r}
i <- 39

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.5)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% arrange(plate.y)
 

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
 

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
The digest of the 2014 sample was in the same plate as APCL14_109.  Cannot eliminate the possibility of lab error.

#i 39 decision
The samples from Palanas are clearly a recapture but lab error cannot be eliminated as a possibility for the Wangag site.
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L1237", issue = "This ligation matches L3037 and L5267 but is at a different site.  Cannot rule out lab error as a possibility because this sample was on the same digest plate as another sample that was collected at Palanas in 2014 that has not be genotyped.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% c("APCL15_402189", "APCL16_574"), 1514, gen_id))
```
# i 41 - same year, shrinkage

```{r}
i <- 41

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 41 - shrinkage of 0.8cm
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1458, gen_id))
```


# i 42 - no gen_id

```{r}
i <- 42

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# 42 decision
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>%
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```


 # i 43 - pool 92, 93 issue

```{r}
i <- 43

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# Find all matches that are between pool 92 pool 93 samples
```{r}
lab <- read_db("Laboratory")

pools <- lab %>% 
  tbl("ligation") %>% 
  filter(pool == "P092" | pool == "P093") %>% 
  collect()

# find all of the samples that are match that are in pool 92 and pool 93
bad_match <- new_pairs %>% 
  filter(first_id %in% pools$ligation_id & second_id %in% pools$ligation_id)
```
It's only 6 samples, 3 from each pool.

I think the 2 samples from the same site are a match and the other has a lab issue
#i 43 decision
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L4957", issue = "This ligation matches L5005 and has the same barcode number in an adjacent pool but not on the same plate.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```


# i 44 - different sites, multiple samples

```{r}
i <- 44

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.5)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 same_dig <- rbind(same_dig, y_work) %>% arrange(plate.y)
 

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
 

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
There is potential for labwork overlap errors
# decision 44
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = "L0936", issue = "This ligation matches to fish caught in Palanas but this fish was caught in Wangag and lab error cannot be ruled out.")

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)

```
Add the gen_id to the rest of them

```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id & sample_id != "APCL14_450", 557, gen_id))

# test
fish %>% 
  filter(sample_id %in% y$sample_id)

# remove gen_id from the ligaiton with issues
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL14_450", NA, gen_id))

rm(bad_match)
```

 i 45 - 47 same group
# i 48 - shrinkage

```{r}
i <- 48

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 48 - shrinkage of 0.4cm
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1427, gen_id))
```


# i 49 - same year

```{r}
i <- 49

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 49
These fish are all at Palanas, 2 from same year, one from a prior year, all caught in the same spot.
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1394, gen_id))
```

i 50 - same group

# i 51 - same year
 a month apart
```{r}
i <- 51

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1662, gen_id))
```


# i 52 - no gen_ids
```{r}
i <- 52

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 52
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 53 - different sites
```{r}
i <- 53

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "N. Magbangon", 
         year(date) == "2014") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.5)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 same_dig <- rbind(same_dig, y_work) %>% arrange(plate.y)
 

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
 

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
Potential labwork overlap on the extraction and digest plates for the 2014 labwork

# site B
```{r}
# get dives
siteb <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2016") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% siteb$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

siteb <- left_join(temp, siteb, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% siteb$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

siteb <- left_join(temp, siteb, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- siteb %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% siteb$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

siteb <- left_join(siteb, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(siteb[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

siteb <- siteb %>% 
  mutate(distkm = alldists) 

siteb_small <- siteb %>% 
filter(as.numeric(size) < 10.5)

# how many of siteb that are small enough to match our fish are ungenotyped? 
nonb <- siteb_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, siteb, siteb_small)
```

# site B labwork
```{r eval=FALSE}
siteb_work <- work_history(nonb, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- siteb_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- siteb_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% 
   # arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- siteb_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- siteb_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- siteb_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(siteb_work, y_work, nonb)
```
Potential labwork overlap in digest, ligation, plates and barcodes in 2016.  Can't say which is in error.  Leaving both as it but not listing them as a match.

 i 54 - already did this group
 i 55 - same group
 i 56 - same group
# i 57 - shrinkage
```{r}
i <- 57

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
Shrinkage of 0.1 cm
```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 741, gen_id))
```

# i 58 - no gen_ids
```{r}
i <- 58

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 58
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 59 - same year
a month apart
```{r}
i <- 59

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

```{r}
fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, 1476, gen_id))
```

# i 60 - different sites, different sizes
```{r}
i <- 60

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Cabatoan", 
         year(date) == "2013") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.0)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% arrange(plate.y)
 

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
 

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
There are 9 fish from Cabatoan in 2013 that are small enough to be 10.0 cm in 2016 but there was No labwork overlap for the 2013 samples (none on the same plates)

No clownfish were collected from Sitio Tugas in 2016.  

# decision 60
```{r}
fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, "recapture at different site, APCL13_103 to APCL16_152", fish_notes))
```

# i 61 - no gen_ids
```{r}
i <- 61

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 61
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```


# i 62 - no gen_ids, same year, shrinkage
```{r}
i <- 62

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 62
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 63 - no gen_ids
```{r}
i <- 63

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 63
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 64 - no gen ids
```{r}
i <- 64

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 64
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 65 - no gen ids
```{r}
i <- 65

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 65
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

 i 66 - same group
# i 67
```{r}
i <- 67

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# decision 67
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 68
```{r}
i <- 68

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 68
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 69
```{r}
i <- 69

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 69
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# i 70
```{r}
i <- 70

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

 i 71 same group
# i 72 - different sites
```{r}
i <- 72

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```

# site A
```{r}
# get dives
sitea <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Palanas", 
         year(date) == "2015") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% sitea$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

sitea <- left_join(temp, sitea, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% sitea$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

sitea <- left_join(temp, sitea, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- sitea %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% sitea$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

sitea <- left_join(sitea, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(sitea[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

sitea <- sitea %>% 
  mutate(distkm = alldists) 

sitea_small <- sitea %>% 
filter(as.numeric(size) < 10.5)

# how many of sitea that are small enough to match our fish are ungenotyped? 
nona <- sitea_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, sitea, sitea_small)
```

# site A labwork
```{r eval=FALSE}
sitea_work <- work_history(nona, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- sitea_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- sitea_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% arrange(plate.y)
 

same_lig <- sitea_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- sitea_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- sitea_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))
 

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(sitea_work, y_work, nona)
```
Potential labwork overlap for 3 digests and 1 barcode with 2015 fish from Palanas

# site B
```{r}
# get dives
siteb <- leyte %>% 
  tbl("diveinfo") %>% 
  filter(site == "Sitio Baybayon", 
         year(date) == "2015") %>% 
  collect() %>% 
  select(dive_table_id, site, date, gps)

# get anems
temp <- leyte %>% 
  tbl("anemones") %>% 
  filter(dive_table_id %in% siteb$dive_table_id) %>% 
  collect() %>% 
  select(dive_table_id, anem_table_id, anem_id, anem_obs_time, anem_obs)

siteb <- left_join(temp, siteb, by = "dive_table_id")

# get fish
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(anem_table_id %in% siteb$anem_table_id, 
         !is.na(sample_id)) %>% 
  collect()

siteb <- left_join(temp, siteb, by = "anem_table_id")
rm(temp)

# get lat lon

temp <- siteb %>% 
  mutate(anem_obs_time = force_tz(ymd_hms(str_c(date, anem_obs_time, sep = " ")), tzone = "Asia/Manila")) %>% 
  mutate(anem_obs_time = with_tz(anem_obs_time, tzone = "UTC")) %>% 
  mutate(hour = hour(anem_obs_time), 
         minute = minute(anem_obs_time))

lat <- leyte %>%
    tbl("GPX")  %>% 
    mutate(gpx_date = date(time)) %>%
    filter(gpx_date %in% siteb$date) %>% 
    mutate(gpx_hour = hour(time)) %>%
    mutate(minute = minute(time)) %>%
    mutate(second = second(time)) %>%
    select(-time, -second)%>%
    collect() 

# attach the lat lons
temp <- left_join(temp, lat, by = c("date" = "gpx_date", "hour" = "gpx_hour", "minute" = "minute", "gps" = "unit"))

# summarize the lat lons
temp <- temp %>% 
  group_by(sample_id) %>% 
  summarise(lat = mean(as.numeric(lat)), 
            lon = mean(as.numeric(lon)))

siteb <- left_join(siteb, temp, by = "sample_id")
rm(temp, lat)

# calculate the distances
alldists <- fields::rdist.earth(as.matrix(siteb[,c("lon", "lat")]), as.matrix(investigate[i,c("second_lon", "second_lat")]), miles=FALSE, R=6371) 

siteb <- siteb %>% 
  mutate(distkm = alldists) 

siteb_small <- siteb %>% 
filter(as.numeric(size) < 10.5)

# how many of siteb that are small enough to match our fish are ungenotyped? 
nonb <- siteb_small %>%
  filter(is.na(gen_id)) %>% 
   select(sample_id, size, color, lat, lon, gen_id, distkm)

# rm(alldists, siteb, siteb_small)
```

# site B labwork
```{r eval=FALSE}
siteb_work <- work_history(nonb, "sample_id")

y_work <- work_history(y, "sample_id")

same_ext <- siteb_work %>% 
  filter(plate.x %in% y_work$plate.x)

same_dig <- siteb_work %>% 
  filter(plate.y %in% y_work$plate.y)
 # same_dig <- rbind(same_dig, y_work) %>% 
   # arrange(plate.y)
 # this catches all of the future "sames"

same_lig <- siteb_work %>% 
  filter(plate %in% y_work$plate, 
         !is.na(plate))

same_barcode <- siteb_work %>% 
  filter(barcode_num %in% y_work$barcode_num, 
         !is.na(barcode_num))

same_pool <- siteb_work %>% 
  filter(pool %in% y_work$pool, 
         !is.na(pool))

# rm(same_ext, same_dig, same_lig, same_pool, same_barcode)
# rm(siteb_work, y_work, nonb)
```

12 extractions on the same plate, 2 digests, 2 ligations, 2 pools, plenty of opportunity for lab error.
# decision
```{r}
# backup
iss <- lb %>% 
  tbl("known_issues") %>% 
  collect()
#write_csv(iss, paste0("../db_backups/", Sys.Date(), "known_issues.csv"))

iss <- tibble(ligation_id = c("L2837", "L3151"), issue = c("This ligation matches L3151 but is at a different site, different size, same year.  These samples could have come into contact with other during extraction.", "This ligation matches L2837 but is at a different site, different size, same year.  These samples could have come into contact with each other during extraction."))

#dbWriteTable(lb, "known_issues", iss, row.names=F, append = T)
```
## i 73 - no gen ids
3 fish, 2 are regenotypes, 1 is an identity match to the other 2 ligations
```{r}
i <- 73

  x <- investigate %>% 
    filter(first_id == investigate$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year) 
  
  investigate$mismatch_prop[i]
  rm(x, first, second)
```
# decision 73 - no gen_ids
```{r}
max_gen <- fish %>% 
  summarise(max = max(gen_id, na.rm = T)) %>% 
  collect()

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id %in% y$sample_id, max_gen$max + 1, gen_id))
```

# Starting over on 2019-01-02 to make sure all decisions have been recorded properly and to use new functions

## 1
```{r}
i <- 1
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
```
This is 3 fish, 2 caught in the same year that are clearly genetic recaptures based on size and location and a third that is from a different site and should be evaluated for labwork.

# Send the fish that appear to be good recaptures based on size and distance to the database
```{r}
z <- y %>% 
  filter(site == "Palanas")

temp <- z %>% 
  filter(!is.na(gen_id)) %>% 
  distinct(gen_id)
# there is no gen_id

z <- create_genid(z)

change_db_gen_id(z)

rm(temp, z)
```
Now compare the labwork sample that is from another site

# site A - which fish were caught in the second site in the first year
```{r}
# nona <- meta_site_a(sitea = y$site[2], yeara = year(y$date[1]), sizea = y$size[2])
# nrow(nona)
```

# site A labwork - were any of those fish on the same plates as our match?
Check to see if same_barcode, same_dig, same_ext, same_lig, same_pool are all zero
```{r}
test <- lab_site_a(nona)
nrow(same_barcode)
nrow(same_dig)
nrow(same_ext)
nrow(same_lig)
nrow(same_pool)
```
APCL14_109 has the same barcode (42) as one these 3 but not our 2014 sample (37) - no potential labwork overlap in 2014 samples, move the 2014 sample to known_issues
```{r}
temp <- filter(y_work, grepl("APCL14", sample_id))
temp$ligation_id

# already in the known issues table

```
## 2 same group of fish
## 3
```{r}
i <- 3
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Fish shrinks 0.4 cm, within acceptable limit, assign gen_id in db
```{r}
# make all fish in y have same gen_id
y <- y %>% 
  mutate(gen_id = gen_id[1])
y$gen_id

change_db_gen_id(y)
```
## 4
```{r}
i <- 4
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Suspicious that the fish are exactly the same size, 2 years apart
Get site A info
```{r}
# nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
# temp <- lab_site_a(nona)
# rm(temp)
```
No labwork overlap

Get site B info
```{r eval=FALSE}
nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
temp <- lab_site_b(nonb)

# 3 barcodes
temp <- rbind(same_barcode, y_work)

rm(temp)

```
APCL16_254 has the same barcode as some other fish that were caught in the area of the 2014 fish and there is a possible lab mixup. Add L5019 to the known issues table
```{r}
temp <- tibble(ligation_id = "L5019", issue = "This ligation matches to L0621, but that sample is at another site and is the same size 2 years earlier.  L5019 has the same barcode as 2 other samples that were captured near the location that L0621 was caught, L5211 and L4683 so lab error cannot be ruled out.")

# lb <- write_db("Laboratory")

# #dbWriteTable(lb, "known_issues", temp, append = T, row.names = F)
# dbDisconnect(lb)
# rm(lb)

rm(temp, same_barcode, same_dig, same_ext, same_lig, same_pool, nona, nonb)
```
## 5
```{r}
i <- 5
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Already dealt with this one
## 6
```{r}
i <- 6
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
APCL15_403391 and APCL16_261 appear to be a good match.  APCL16_207 and APCL16_261 are a questionable match.

Assign the gen_id for the good match first
```{r}
# save the original match table
z <- y

# remove the questionable match from y
y <- filter(y, site == "Wangag")
# make all fish in y have same gen_id
y <- y %>% 
  mutate(gen_id = gen_id[1])
y$gen_id

change_db_gen_id(y)

# revert y to original match table
y <- z
rm(z)
```
Examine the questionable match between 2 fish in the same year at different sites 
```{r}
# remove the "good match" from the table to examine the questionable match in more detail
# y <- filter(y, year == "16")
# 
# nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
# temp <- lab_site_a(nona)
```
There is a lot of potential for labwork overlap (~36 possible samples).  Move the San Agustin fish to the known issues table - already there as L4978

## 7 same group
## 8
```{r}
i <- 8
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Same group but one more fish from 2014
Check the updated gen_id on the 2016 fish
```{r}
temp <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id == "APCL16_261") %>% 
  select(sample_id, gen_id) %>% 
  collect()
```
1086, matches the 2014 fish.
## 9 same group
## 10 same group
## 11 
```{r}
i <- 11
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Different sites, check labwork
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
rm(temp)
```
no overlapping labwork from the 2013 fish
```{r eval=FALSE}
nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
temp <- lab_site_b(nonb)

temp <- rbind(same_pool, y_work)
rm(temp)


```
There are 4 possible samples that could've been mixed up with the 2016 fish that were captured near the 2013 fish, adding the 2016 fish to the known issues - already there
## 12
```{r}
i <- 12
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Fish was caught in the same year in virtually the same spot.  Appears to be a good match
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 13
```{r}
i <- 13
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Fish caught in the same year at different sites with shrinkage.  Check lab work
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
rm(temp)
```
Up to 23 possible candidates for labwork overlap, also these 2 samples were on the same plate and used the same barcode even though they were in pools 92 and 93.  Already updated known issues

## 14
```{r}
i <- 14
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
```
Fish shrinks, but only by 0.8cm.  Adding gen_id as match
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])

change_db_gen_id(y)
```
## 15
```{r}
i <- 15
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
This fish did not appear as a match in the past but was captured in the same spot and the growth is acceptable. Note the high mismatch proportion.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 16
```{r}
i <- 16
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
This fish did not appear as a match in the past but was captured in the same spot and the growth is acceptable. Note the low mismatch proportion.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
  
```
## 17
```{r}
i <- 17
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
This fish was caught in the same year at different sites with very different sizes.  Check lab work
```{r eval=FALSE}
nona <- select(y, sample_id, size, color, lat, lon, gen_id)
temp <- lab_site_a(nona)
rm(temp)
```
Same extraction plate, same ligation plate, same pool. Already noted in known issues.

## 18
```{r}
i <- 18
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish shrinks by 0.4cm, caught in same place.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 19
```{r}
i <- 19
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish caught at same spot, growth is normal.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 20 same group
## 21
```{r}
i <- 21
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish caught at same spot, growth is normal.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 22 same group
## 23
```{r}
i <- 23
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish moved sites, check labwork
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
No labwork overlap on 2014 fish
```{r eval=FALSE}
nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
temp <- lab_site_b(nonb)
```
No labwork overlap on 2015 fish.  It appears this fish moved sites but we are not prepared to accept that so have to make a note for the fish.
```{r}
# backup database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()
#write_csv(fish, paste0("../db_backups/", Sys.time(), "_clownfish.csv"))

fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, ifelse(is.na(fish_notes), "APCL15_376136 and APCL14_408 return as genetic recaptures but were captured at Wangag and Palanas.  There is no overlapping labwork and the proportion of mismatch is 0.06", paste("APCL15_376136 and APCL14_408 return as genetic recaptures but were captured at Wangag and Palanas.  There is no overlapping labwork and the proportion of mismatch is 0.06", fish_notes, sep = ", ")), fish_notes))

# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, overwrite = T, row.names = F)
# dbDisconnect(ley)
rm(ley, fish, nona, nonb, same_barcode, same_dig, same_ext, same_lig, same_pool, temp, y, y_work)
```
## 24
```{r}
i <- 24
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Three fish, 2 are clearly a match based on lat lon and size, the third needs to be checked against lab work. 
Write the good match to the db now
```{r}
z <- y
y <- filter(y, site == "Palanas")
y <- create_genid(y)
change_db_gen_id(y)

y <- z
rm(z)
```
Check the labwork for the questionable match
```{r eval=FALSE}

nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
There is potential for lab overlap on the digest plate.  Moving the 2014 fish to the known issues table - already there.
## 25 same group
## 26
```{r}
i <- 26
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
3 Fish all caught in the same place, growth looks good.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 27 same fish
## 28 same fish
## 29
```{r}
i <- 29
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish caught in the same spot, growth is acceptable
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 30 same group
## 31
```{r}
i <- 31
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish was captured in the same place and shrinks by 0.5cm, within measurement error.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 32
```{r}
i <- 32
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
These fish were caught in the same spot some time apart.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 33
```{r}
i <- 33
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
The 2 fish from Wangag were caught in the same spot and appear to be a good match based on size and lat lon.  Update db
```{r}
z <- y
y <- filter(y, site == "Wangag")
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
Check labwork for other fish
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
No labwork overlap.  Making a note on the samples.
```{r}
# backup database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()
#write_csv(fish, paste0("../db_backups/", Sys.time(), "_clownfish.csv"))

fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, ifelse(is.na(fish_notes), "APCL14_547, APCL15_404287, and APCL16_800 return as genetic recaptures but were captured at Wangag and Palanas.  There is no overlapping labwork and the proportion of mismatch is 0.04", paste("APCL14_547, APCL15_404287, and APCL16_800 return as genetic recaptures but were captured at Wangag and Palanas.  There is no overlapping labwork and the proportion of mismatch is 0.04", fish_notes, sep = ", ")), fish_notes))

# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, overwrite = T, row.names = F)
# dbDisconnect(ley)
rm(ley, fish, nona, same_barcode, same_dig, same_ext, same_lig, same_pool, temp, y, y_work)
```
## 34 same group
## 35
```{r}
i <- 35
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish caught in the same spot, good growth
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 36 same group
## 37
```{r}
i <- 37
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish shrinks but was captured in the same spot and the difference in size is only 0.5cm.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 38
```{r}
i <- 38
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish moved sites, check labwork
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
No labwork overlap
```{r eval=FALSE}
nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
temp <- lab_site_b(nonb)
```
Some potential for the 2016 fish to have labwork overlap.  Moving this fish to known issues
```{r eval=FALSE}
temp <- tibble(ligation_id = "L5050", issue = "This ligation matches to L0687, but this sample was captured at Elementary School and that sample was captured at Poroc San Flower.  There are seven possible alternate samples in the extraction, digest, and ligation plates as well as 4 in the same pool.")

# lb <- write_db("Laboratory")
# 
# #dbWriteTable(lb, "known_issues", temp, append = T, row.names = F)
# dbDisconnect(lb)
# rm(lb)

rm(temp, same_barcode, same_dig, same_ext, same_lig, same_pool, nona, nonb)
```
## 39
```{r eval=FALSE}
i <- 39
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Three fish, 2 appear to be a good match based on size and lat lon and have already been looked at up above, the third from 2014 and is in the known issues table.
## 40 same group
## 41
```{r eval=FALSE}
i <- 41
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish caught in same spot, shrinkage is within allowable limits.
```{r eval=FALSE}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 42
```{r eval=FALSE}
i <- 42
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish needs gen_id
```{r eval=FALSE}
y <- create_genid(y)

change_db_gen_id(y)
```
## 43
```{r eval=FALSE}
i <- 43
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish captured at 2 very distant sites
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
no overlap on the 2016 samples
```{r eval=FALSE}
# nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
# temp <- lab_site_b(nonb)
```
The same barcode was used for the 2017 samples but no other overlap occurs.  These 2 samples were found in pools 92 and 93 with the same barcode.  A note has been made in known issues.
## 44
```{r eval=FALSE}
i <- 44
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Four fish, captured at 2 sites, the fish from 2014 was captured at Wangag, the rest are at Palanas and based on lat lon and size appear to be good matches.
```{r eval=FALSE}
z <- y
y <- filter(y, site == "Palanas")
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
Check the labwork for the 2014 sample
```{r eval=FALSE}
y <- z
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
There are opportunities for labwork overlap on the digest, ligation plates, with barcodes and pools for at least 1 sample but possibly 3 candidates in digest plate. Already noted in the known issues.  Interestingly, APCL14_450 was regenotyped and does not match to this ligation.
## 45-47 same group
## 48
```{r eval=FALSE}
i <- 48
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 0.4cm of shrinkage is within allowable limits.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 49
```{r}
i <- 49
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
 
## 50 same group
## 51
```{r}
i <- 51
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 52
```{r}
i <- 52
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 53
```{r}
i <- 53
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Check labwork
```{r eval=FALSE}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
There are 7 potential samples that could've been mixed up with the 2014 sample on an extraction plate.
```{r eval=FALSE}
nonb <- meta_site_b(y$site[1], y$year[2], y$size[2])
temp <- lab_site_b(nonb)
```
There is 1 sample on a digest plate and 2 samples on a ligation plate that could've been mixed up with the 2016 sample.  Making a note for both samples
```{r eval=FALSE}
temp <- tibble(ligation_id = c("L0974", "L5107"), issue = c("This ligation matches to L5107, but this sample was captured at Palanasand that sample was captured at N. Magbangon.  There are seven possible alternate samples in the extraction plate.", "This ligation matches to L0974, but that sample was captured at Palanas and this sample was captured at N. Magbangon.  There are one or two possible alternate samples in the digest and ligation plates."))

# lb <- write_db("Laboratory")
# 
# #dbWriteTable(lb, "known_issues", temp, append = T, row.names = F)
# dbDisconnect(lb)
# rm(lb)

rm(temp, same_barcode, same_dig, same_ext, same_lig, same_pool, nona, nonb)
```
## 54
```{r}
i <- 54
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Based on size and location these look like genetic recaptures
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 55-56 same group
## 57
```{r}
i <- 57
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Based on size and location these look like genetic recaptures. 0.1cm shrinkage is within allowable parameters.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 58
```{r}
i <- 58
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 59
```{r}
i <- 59
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Based on size and location these look like genetic recaptures. 0.5cm shrinkage is within allowable parameters.
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 60
```{r}
i <- 60
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Fish changes site and shrinks over 3 years. Check labwork.
```{r}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
No labwork overlap of 2013 samples.  We didn't capture any fish at Sitio Tugas in 2016 so there are no candidates for lab error in that year. Making a note in the clownfish table because these were different sites.

No labwork overlap.  Making a note on the samples.
```{r}
# backup database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()
#write_csv(fish, paste0("../db_backups/", Sys.time(), "_clownfish.csv"))

fish <- fish %>% 
  mutate(fish_notes = ifelse(sample_id %in% y$sample_id, ifelse(is.na(fish_notes), "APCL13_103 and APCL16_152 return as genetic recaptures but were captured at Sitio Tugas and Cabatoan.  There is no overlapping labwork and the proportion of mismatch is 0.03", paste("APCL13_103 and APCL16_152 return as genetic recaptures but were captured at Sitio Tugas and Cabatoan.  There is no overlapping labwork and the proportion of mismatch is 0.03", fish_notes, sep = ", ")), fish_notes))

# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, overwrite = T, row.names = F)
# dbDisconnect(ley)
rm(ley, fish, nona, same_barcode, same_dig, same_ext, same_lig, same_pool, temp, y, y_work)
```
## 61
```{r}
i <- 61
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 62
```{r}
i <- 62
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 63
```{r}
i <- 63
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 64
```{r}
i <- 64
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 65
```{r}
i <- 65
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 66 same group
## 67
```{r}
i <- 67
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 68
```{r}
i <- 68
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 69
```{r}
i <- 69
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 70
```{r}
i <- 70
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 71 same group
## 72
```{r}
i <- 72
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Fish caught in the same year at different sites, check lab work
```{r}
nona <- meta_site_a(y$site[2], y$year[1], y$size[2])
temp <- lab_site_a(nona)
```
These samples were on the same extract plate in A6 and B1.  There are also 3 candidates for mixup on a digest plate.  Cannot rule out lab error.  Adding these samples to known issues - already there.
## 73
```{r}
i <- 73
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = 1260)
change_db_gen_id(y)
```
## 74 same group
## 75
```{r}
i <- 75
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
# restarting at 1 because I auto-assigned all of the new matches that needed a gen_id
## 1 - 13 already done
## 14
```{r}
i <- 14
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id) %>% 
  collect()
test$gen_id

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
test$ligation_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 15-23 already done
## 24
```{r}
i <- 24
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
test$ligation_id
```
2 fish are a genetic recapture based on size and lat lon, 1 fish is questionable and is on the known issues list
```{r}
z <- y
y <- filter(y, site == "Palanas")
y <- create_genid(y)
change_db_gen_id(y)
```
## 25 same group
## 26
```{r}
i <- 26
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 27-28 same group
## 29-58 already done
## 59
```{r}
i <- 59
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
Appears to be a genetic recapture based on size and lat lon. 
```{r}
y <- y %>% 
  mutate(gen_id = 1672)
change_db_gen_id(y)
```
## 60 already done
## 61
```{r}
i <- 61
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- y %>% 
  mutate(gen_id = 1676)
change_db_gen_id(y)
```
## 62-66 already done
## 67
```{r}
i <- 67
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 68
```{r}
i <- 68
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 69 already done
## 70
```{r}
i <- 70
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 71
```{r}
i <- 71
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- y %>% 
  mutate(gen_id = 1678)
change_db_gen_id(y)
```
## 72 same group
## 73
```{r}
i <- 73
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- y %>% 
  mutate(gen_id = 1678)
change_db_gen_id(y)
```
## 74 same group
## 75-76 already done
```{r}
i <- 77
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```


```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 78 already done
## 79
```{r}
i <- 79
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 80
```{r}
i <- 80
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 81
```{r}
i <- 81
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 82
```{r}
i <- 82
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- y %>% 
  mutate(gen_id = gen_id[1])
change_db_gen_id(y)
```
## 83 same group
## 84 already done
## 85
```{r}
i <- 85
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 86
```{r}
i <- 86
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 87-89 already done

# Double check the matched fish to make sure they look right
```{r}
gen_recaps <- leyte %>% 
  tbl("clownfish") %>%
  group_by(gen_id) %>% 
  summarize(caps = n()) %>% 
  filter(caps > 1, 
         !is.na(gen_id)) %>% 
  collect()
```
Gen_id 1678 has 160 recaps, something is wrong here
```{r}
wrong <- leyte %>% 
  tbl("clownfish") %>%
  filter(gen_id == 1678) %>% 
  collect()
```
Remove 1678 from the database and then examine idcsv and remove any pairs that already have a gen_id, then re-examine these matches.

```{r}
# fish <- leyte %>% 
#   tbl("clownfish") %>% 
#   collect()
# 
# change <- fish %>% 
#   filter(gen_id == 1678) %>% 
#   mutate(gen_id = NA)
# 
# fish <- anti_join(fish, change, by = "fish_table_id")
# 
# fish <- rbind(fish, change)
# 
# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, overwrite = T, row.names = F)
# dbDisconnect(ley) 
rm(ley)

# which fish are already accepted genetic recaptures?
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")
idcsv <- idcsv %>% 
  select(-first_gen_id, -second_gen_id)
first <- fish_anem_dive(idcsv$first_sample_id, leyte) %>% 
  select(sample_id, gen_id) %>% 
  rename(first_gen_id = gen_id)
  
second <- fish_anem_dive(idcsv$second_sample_id, leyte) %>% 
  select(sample_id, gen_id) %>% 
  rename(second_gen_id = gen_id)

idcsv <- left_join(idcsv, first, by = c("first_sample_id" = "sample_id"))
idcsv <- left_join(idcsv, second, by = c("second_sample_id" = "sample_id"))

old_pairs <- idcsv %>% 
  filter(first_gen_id == second_gen_id)

new_pairs <- anti_join(idcsv, old_pairs)

# remove samples with known issues
ki <- lab %>% 
  tbl("known_issues") %>% 
  collect()

# remove any first_id's that are on the known issues list
new_pairs <- anti_join(new_pairs, ki, by = c("first_id" = "ligation_id"))
new_pairs <- anti_join(new_pairs, ki, by = c("second_id" = "ligation_id"))

```
# auto assign any that will go
```{r}
investigate <- tibble()
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

for(i in seq(new_pairs$first_id)){
  print(i)
  x <- new_pairs %>% 
    filter(first_id == new_pairs$first_id[i])
  first <- x %>% 
    select(contains("first"))
  second <- x %>% 
    select(contains("second"))
  names(first) <- substr(names(first), 7, 20)
  names(second) <- substr(names(second), 8, 25)
  y <- rbind(first, second) %>% 
    distinct() %>% 
    mutate(year = substr(sample_id, 5,6), 
           size = as.numeric(size)) %>% 
    arrange(year)
  
  temp <- y %>% 
    filter(is.na(fish_table_id))
  if (nrow(temp) == 0){
    if (length(unique(y$year)) == nrow(y)){
      if (!is.unsorted(y$size)){
        if (nrow(distinct(y,site)) == 1){
          temp <- y %>% 
            filter(!is.na(gen_id)) %>% 
            distinct(gen_id)
          if (nrow(temp) == 1){
            fish <- fish %>%
               mutate(gen_id = ifelse(sample_id %in% y$sample_id, temp$gen_id, gen_id))
          
           }else{
              temp <- y %>% 
              distinct(gen_id)
            if (nrow(temp) == 1 & is.na(temp$gen_id)){
              y <- create_genid(y)
              fish <- fish %>%
               mutate(gen_id = ifelse(sample_id %in% y$sample_id, y$gen_id[1], gen_id))
            }else{
                z <- new_pairs %>%
              filter(first_id == new_pairs$first_id[i]) %>% 
              mutate(id_note = "More than one gen_id!")
            investigate <- rbind(investigate, z) %>%
              distinct()
              }
            
            }
        }else{
          z <- new_pairs %>%
            filter(first_id == new_pairs$first_id[i]) %>% 
            mutate(id_note = "Fish moved sites")
          investigate <- rbind(investigate, z) %>%
            distinct()
        }
      }else{
        z <- new_pairs %>%
          filter(first_id == new_pairs$first_id[i]) %>% 
          mutate(id_note = "Fish shrinks!")
        investigate <- rbind(investigate, z) %>%
          distinct()
      }
    }else{
      z <- new_pairs %>%
        filter(first_id == new_pairs$first_id[i]) %>% 
        mutate(id_note = "Fish caught in same year!")
      investigate <- rbind(investigate, z) %>%
        distinct()
    }
  }else{
    z <- new_pairs %>%
      filter(first_id == new_pairs$first_id[i]) %>% 
      mutate(id_note = "Fish had previous identity issues!")
    investigate <- rbind(investigate, z) %>%
      distinct()
  }
}

investigate <- investigate %>% 
  mutate(first_tag_id = as.character(first_tag_id), 
         second_tag_id = as.character(second_tag_id))
  

```
Seven samples need investigation, the first is a set of samples that are from different sites but have no lab overlap, 2 is the same, 3 is the same
## 1 - 3 are done
## 4
```{r}
i <- 4
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```


## 5 same group
## 6
```{r}
i <- 6
investigate$id_note[i]
investigate$mismatch_prop[i]
y <- check_id_match(i, investigate)
y$sample_id
y$size
y$color
y$site
y$gen_id
y$lat
y$lon
y$recap
y$tag_id

# have I already matched these?
test <- leyte %>% 
  tbl("clownfish") %>% 
  filter(sample_id %in% y$sample_id) %>% 
  select(sample_id, gen_id, fish_notes) %>% 
  collect()
print("gen_id")
test$gen_id
test$fish_notes

a <- lig_from_samp(y$sample_id)
test <- lab %>% 
  tbl("known_issues") %>% 
  filter(ligation_id %in% a$ligation_id) %>% 
  collect()
print("issues")
test$ligation_id
```
```{r}
y <- create_genid(y)
change_db_gen_id(y)
```
## 7 same group

# Remove gen_id from samples with known issues
```{r}
# Pull in the known issues table if needed

ki <- lab %>% 
  tbl("known_issues") %>% 
  collect()

iss <- samp_from_lig(ki)

# get the work history for the sample ids to see if there are other ligations that might have been successful.
iss <- work_history(iss, "sample_id") %>% 
  filter(grepl("P", pool))

# which of these samples has only one pool?
count <- iss %>% 
  group_by(sample_id) %>% 
  summarise(count = n()) %>% 
  filter(count == 1) %>% 
  mutate(gen_id = NA) %>% 
  distinct()

# remove the gen_id for these samples
 change_db_gen_id(count)

 # test to make sure it worked - looks good
 test <- leyte %>% 
   tbl("clownfish") %>% 
   filter(sample_id %in% count$sample_id) %>% 
   select(sample_id, gen_id) %>% 
   collect()
```


# EVERYTHING AFTER THIS POINT IS FOR REMOVING THE MATCHES FROM THE GENEPOP SO IT CAN CONTINUE FOR PARENTAGE ANALYSIS.  FOR CONTINUED ID ANALYSIS, OPEN id_process.R

## Open genepop ------------------------------------------------------------ -->
```{r}
# this should be the genepop you used as input for Cervus ID
genfile <- "data/seq03-33_identity/seq33_03_baits_only_SNPs.gen"
genedf <- read_genepop(genfile)

# TEST - make sure the first 2 columns are names and a contig and get number of rows
names(genedf[,1:2]) # [1] "names" "dDocent_Contig_11_59"
nrow(genedf) # 2893
```

# Calculate the number of loci for analysis -------------------------------
```{r}
# convert 0000 to NA in the genepop data
genedf[genedf == "0000"] <- NA
# TEST - make sure there are no "0000" left
which(genedf == "0000") # should return integer(0)

# count the number of loci per individual - takes 2-7 minutes
Sys.time()
for(i in seq(genedf$names)){
  genedf$numloci[i] <- sum(!is.na(genedf[i,]))
}
Sys.time()

### WAIT ###

# TEST - make sure all of the numloci were populated
which(is.na(genedf$numloci)) # should return integer(0)

```
# determine which to remove based on num loci of each id match --------
```{r}
# fix names of samples
genedf <- genedf %>%
  mutate(names = substr(names, 6, 25)) %>% 
  rename(id = names)

# fix samples that have a .F at the end
genedf <- genedf %>% 
  mutate(id = ifelse(grepl(".F", id), str_replace(id, ".F", ""), id))

# figure out which rows of idcsv to keep
accepted_matches <- leyte %>% 
  tbl("clownfish") %>%
  group_by(gen_id) %>% 
  summarize(caps = n()) %>% 
  filter(caps > 1, 
         !is.na(gen_id)) %>% 
  collect() 

accepted_matches <- leyte %>% 
  tbl("clownfish") %>% 
  select(sample_id, gen_id) %>% 
  filter(gen_id %in% accepted_matches$gen_id) %>% 
  collect() %>% 
  mutate(accepted = T)

  # get the work history for the matches
pairs <- work_history(accepted_matches, "sample_id")

# remove any work that is not in the current genepop
pairs <- pairs %>% 
  filter(ligation_id %in% genedf$id)
  
# bring in the big table from the identity analysis
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")
result <- tibble()
for(i in seq(pairs$sample_id)){
x <- idcsv %>% 
  filter(first_id %in% pairs$ligation_id[i])
if (nrow(x) == 0){
  x <- idcsv %>% 
  filter(second_id == pairs$ligation_id[i])
}
 # a & b are  the line numbers from genepop file that matches an the first and second ID in the match table
  a <- which(genedf$id %in% x$first_id)
  b <- which(genedf$id %in% x$second_id)
  # if the first num loci is greater than or equal to the second num loci, drop the second id, or else drop the first.
x <- x %>% 
  mutate(drop = ifelse(genedf$numloci[a] > genedf$numloci[b], second_id, first_id))
result <- rbind(result, x)

 }
result <- result %>% 
  select(drop) %>% 
  distinct()

genedf <- genedf %>%
  filter(!id %in% result$drop)
```
# Look for regenotypes again:
```{r}
# select only the id column from the genedf
ids <- genedf %>% 
  select(id) %>% 
  rename(ligation_id = id)

# get sample ids 
ids <- samp_from_lig(ids)

# find any sample ids that occur more than once
regenos <- ids %>% 
  group_by(sample_id) %>% 
  summarise(count = n()) %>% 
  filter(count > 1)

# find the ligation ids that go with them
regenos <- left_join(regenos, ids, by = "sample_id")

# keep only the regeno with the highest number of loci for further analysis (in the future these files will be merged)
result <- tibble()
for(i in seq(regenos$sample_id)){
x <- regenos %>% 
  filter(sample_id == regenos$sample_id[i])
 # a & b are  the line numbers from genepop file that matches an the first and second ID in the match table
 x <- left_join(x, select(genedf, id, numloci), by = c("ligation_id" = "id")) %>% 
   arrange(numloci)
  # if the first num loci is greater than or equal to the second num loci, drop the second id, or else drop the first.
x <- x %>% 
  mutate(drop = ligation_id[1])
result <- rbind(result, x)

 }
result <- result %>% 
  select(drop) %>% 
  distinct()

genedf <- genedf %>%
  filter(!id %in% result$drop)

```
# convert all the NA genotypes to 0000
```{r}
genedf[is.na(genedf)] = "0000"
# TEST - make sure there are no NA's left
which(is.na(genedf)) # should return integer(0)

# remove numloci column
genedf <- select(genedf, -numloci)
```


# Write out genepop  ------------------------------------------------------
```{r}
# Build the genepop components
msg <- c("This genepop file was generated using a script called identity_analysis.R written by Michelle Stuart with help from Malin Pinsky and Ryan Batt")

loci <- paste(names(genedf[,2:ncol(genedf)]), collapse =",")

gene <- vector()
sample <- vector()

Sys.time()
for (i in seq(genedf$id)){
  gene[i] <- paste(genedf[i,2:ncol(genedf)], collapse = " ")
  sample[i] <- paste(genedf[i,1], gene[i], sep = ", ")
}
Sys.time()
  ### WAIT ###

out <- c(msg, loci, 'pop', sample)

# write.table(out, file = "data/seq03-33_norecap.gen", row.names=FALSE, quote=FALSE, col.names=FALSE)
```


```{r}
# cleanup
rm(accepted_matches, change, count, first, fish, gen_recaps, idcsv, ids, investigate, iss, ki, lose, missing, new_pairs, old_pairs, pairs, result, second, temp, test, x, y, z, a, b, i, loci, msg, out, sample)
```

# Assign gen_ids to all of the samples in the genepop that do not currently have any
```{r}
# This will include regenotypes since it is by sample_id instead of ligation id

# Check to see if gen_ids already exist
# full fish database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

# just sample ids and gen ids
genids <- fish %>% 
  select(sample_id, gen_id)

# get sample Ids for genedf 
ids <- genedf %>% 
  select(id) %>% 
  rename(ligation_id = id)

ids <- samp_from_lig(ids) %>% 
  distinct(sample_id)

# Iterate through all of the samples, this takes a while and was a horrible idea.  Next time make a table and write it once.

  # change_db_gen_id(y)

# was taking so long I stopped it

# restore backup

# imported from csv into sequel pro and it imported NAs as 0 for gen_id
clown <- fish
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

# change 0 gen_id to NA
fish <- fish %>% 
  mutate(gen_id = ifelse(gen_id == 0, NA, gen_id))

# assign gen_ids to the samples in the genedf
ids <- ids %>% 
  mutate(gen_id = 1690:(1689+nrow(ids)))

change <- tibble()
# add the gen_ids to the fish table
for (i in seq(ids$sample_id)){
  x <- fish %>%
    filter(sample_id == ids$sample_id[i]) %>% 
    mutate(gen_id = ifelse(is.na(gen_id), ids$gen_id[i], gen_id))
  change <- rbind(change, x)
}

# remove the changed rows from the fish table (these rows in the fish table are unchanged)
fish <- anti_join(fish, change, by = "fish_table_id")

# add the changed rows to the fish table
fish <- rbind(fish, change)

# write the fish table to the database
# ley <- write_db("Leyte")
# #dbWriteTable(ley, "clownfish", fish, overwrite = T, row.names = F)
# dbDisconnect(ley)
# rm(ley)
```

# Generate plots of distribution of distance between captures for obvious recaptures and flagged recaptures
```{r plots}
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")

# mark rows where the match was considered an obvious recapture based on genetics, size, lat lon.

# get the most current gen_ids for the first samples 
first <- fish_anem_dive(idcsv$first_sample_id, leyte)
 names(first) <- paste0("first_", names(first)) 
 
 # remove the sample_id from this group of names to use later
 f_except <- select(first, -first_sample_id)
 x <- names(f_except)

# get the most current gen_ids for the second samples
sec <- fish_anem_dive(idcsv$second_sample_id, leyte)
names(sec) <- paste0("second_", names(sec))

# remove the sample_id from this group of names to use later
s_except <- select(sec, -second_sample_id)
y <- names(s_except)

# remove the above columns from idcsv so that the most updated version can be reattached
idcsv <- idcsv %>% 
  select(-x, -y)

idcsv <- left_join(idcsv, first, by = "first_sample_id")
idcsv <- left_join(idcsv, sec, by = "second_sample_id")

# mark all of the rows where the genetic recapture is cofirmed by  size, location
idcsv <- idcsv %>% 
  mutate(mark = ifelse(first_gen_id == second_gen_id, "apparent", "questionable"), 
         mark = ifelse(is.na(mark), "questionable", mark))

dist <- idcsv %>% 
  select(first_id, second_id, distkm, mark) %>% 
  group_by(distkm, mark) %>% 
  summarise(count = n())

# convert to meters
dist <- dist %>%
  mutate(distm = distkm * 1000)


ggplot(dist, aes(distm, fill = mark)) +
  geom_histogram(binwidth = 1000, position = "dodge") +
  scale_y_continuous(limits = c(0, 5))+
  theme_bw()+
  theme(
    plot.background = element_blank()
    ,panel.grid.major = element_blank()
    ,panel.grid.minor = element_blank()
    ,panel.border = element_blank()
  ) 
  
```

Another way to look at it
```{r}

ggplot(dist, aes(x = mark, y = distm)) + 
  geom_boxplot() +
  coord_flip() +
  geom_jitter()
```

Ignore the questionable far distance fish
```{r}
dist_sm <- dist %>% 
  filter(distm < 1000)

ggplot(dist_sm, aes(x = mark, y = distm)) + 
  geom_boxplot() +
  coord_flip() +
  geom_jitter()
```

It seems like a fair chance that these questionable matches that were caught less than 100m from each other are true matches.  Examine them
```{r}
quest <- idcsv %>% 
  filter(first_gen_id != second_gen_id, 
         distkm < 0.1) %>% 
  select(first_sample_id, second_sample_id, first_size, second_size, first_site, second_site, first_gen_id, second_gen_id, first_lat, second_lat, first_lon, second_lon, distkm)
```
There appears to be a gen_id issue with these samples that prevented them from appearing as matches in the database.
```{r}
# make the second_gen_id match the first_gen_id
fix <- quest %>% 
  mutate(second_gen_id = first_gen_id)

# take all of the columns belonging to the first fish and pull them into a table
firsts <- fix %>% 
  # select only the id column for matching and the gen id column that needs to be updated in the db
  select(first_sample_id, first_gen_id) %>% 
  # rename them for one table 
  rename(sample_id = first_sample_id, gen_id = first_gen_id)

# take all of the columns belonging to the first fish and pull them into a table
seconds <- fix %>% 
  # select only the id column for matching and the gen id column that needs to be updated in the db
  select(second_sample_id, second_gen_id) %>% 
  # rename them for one table 
  rename(sample_id = second_sample_id, gen_id = second_gen_id)

# combine into one table
fix_list <- rbind(firsts, seconds) %>% 
  distinct()

# some samples are in the fix_list more than once
dups <- fix_list %>% 
  group_by(sample_id) %>% 
  summarise(count = n()) %>% 
  filter(count > 1)

# make sure every instance of a sample has the same gen_id
for (i in seq(dups$sample_id)){
  x <- fix_list %>% 
    filter(sample_id == dups$sample_id[i])
  y <- fix_list %>% 
    filter(gen_id %in% x$gen_id) %>% 
    distinct()
  fix_list <- fix_list %>% 
    mutate(gen_id = ifelse(sample_id %in% y$sample_id, x$gen_id[1], gen_id)) %>% 
    distinct()
}



# pull in the clownfish table from the database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

# The following line didn't work, some sample_ids were not changed!!!
# fish <- fish %>%
# mutate(gen_id = ifelse(sample_id %in% fix_list$sample_id, fix_list$gen_id, gen_id))

for (i in seq(fix_list$gen_id)){
  fish <- fish %>% 
    mutate(gen_id = ifelse(sample_id == fix_list$sample_id[i], fix_list$gen_id[i], gen_id))
}

# from an earlier db debacle, all of the NULL's in the db table have been covnerted to NA's, attempting to convert them back here. - Doesn't work.

# convert NA to NULL in the genepop data
# fish[fish == NA] <- NULL

 # backup db
  ley <- write_db("Leyte")
  clown <- ley %>% 
    tbl("clownfish") %>% 
    collect()
  #write_csv(clown, path = paste0("../db_backups/", Sys.time(), "_clownfish_db.csv"))
  
    # write change
# #dbWriteTable(ley, "clownfish", fish, row.names = F, overwrite = T)
# dbDisconnect(ley)
```
Try graphs again
# Generate plots of distribution of distance between captures for obvious recaptures and flagged recaptures
```{r plots 2}
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")

# mark rows where the match was considered an obvious recapture based on genetics, size, lat lon.

# get the most current gen_ids for the first samples 
first <- fish_anem_dive(idcsv$first_sample_id, leyte)
 names(first) <- paste0("first_", names(first)) 
 
 # remove the sample_id from this group of names to use later
 f_except <- select(first, -first_sample_id)
 x <- names(f_except)

# get the most current gen_ids for the second samples
sec <- fish_anem_dive(idcsv$second_sample_id, leyte)
names(sec) <- paste0("second_", names(sec))

# remove the sample_id from this group of names to use later
s_except <- select(sec, -second_sample_id)
y <- names(s_except)

# remove the above columns from idcsv so that the most updated version can be reattached
idcsv <- idcsv %>% 
  select(-x, -y)

idcsv <- left_join(idcsv, first, by = "first_sample_id")
idcsv <- left_join(idcsv, sec, by = "second_sample_id")

# mark all of the rows where the genetic recapture is cofirmed by  size, location
idcsv <- idcsv %>% 
  mutate(mark = ifelse(first_gen_id == second_gen_id, "apparent", "questionable"), 
         mark = ifelse(is.na(mark), "questionable", mark))

dist <- idcsv %>% 
  select(first_id, second_id, distkm, mark) %>% 
  group_by(distkm, mark) %>% 
  summarise(count = n())

# convert to meters
dist <- dist %>%
  mutate(distm = distkm * 1000)

```

Ignore the questionable far distance fish
```{r}
dist_sm <- dist %>% 
  filter(distm < 1000)

ggplot(dist_sm, aes(x = mark, y = distm)) + 
  geom_boxplot() +
  coord_flip() +
  geom_jitter()
```
What fish are left at less than 100m?
```{r}
quest <- idcsv %>% 
  filter(first_gen_id != second_gen_id, 
         distkm < 0.1) %>% 
  select(first_sample_id, second_sample_id, first_size, second_size, first_site, second_site, first_gen_id, second_gen_id, first_lat, second_lat, first_lon, second_lon, distkm)
```
Not sure why this one fish didn't change.
```{r}
# do these fish have any other matches?
temp <- fish %>% 
  filter(gen_id %in% quest$first_gen_id | gen_id %in% quest$second_gen_id)

temp <- temp %>% 
  mutate(gen_id = 1689)

# pull in the clownfish table from the database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

# The following line didn't work, some sample_ids were not changed!!!
# fish <- fish %>%
# mutate(gen_id = ifelse(sample_id %in% fix_list$sample_id, fix_list$gen_id, gen_id))

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL15_375493", 1689, gen_id))


 # backup db
  ley <- write_db("Leyte")
  clown <- ley %>% 
    tbl("clownfish") %>% 
    collect()
  #write_csv(clown, path = paste0("../db_backups/", Sys.time(), "_clownfish_db.csv"))
  
    # write change
# #dbWriteTable(ley, "clownfish", fish, row.names = F, overwrite = T)
# dbDisconnect(ley)
```

Try graphs again
# Generate plots of distribution of distance between captures for obvious recaptures and flagged recaptures
```{r plots 3}
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")

# mark rows where the match was considered an obvious recapture based on genetics, size, lat lon.

# get the most current gen_ids for the first samples 
first <- fish_anem_dive(idcsv$first_sample_id, leyte)
 names(first) <- paste0("first_", names(first)) 
 
 # remove the sample_id from this group of names to use later
 f_except <- select(first, -first_sample_id)
 x <- names(f_except)

# get the most current gen_ids for the second samples
sec <- fish_anem_dive(idcsv$second_sample_id, leyte)
names(sec) <- paste0("second_", names(sec))

# remove the sample_id from this group of names to use later
s_except <- select(sec, -second_sample_id)
y <- names(s_except)

# remove the above columns from idcsv so that the most updated version can be reattached
idcsv <- idcsv %>% 
  select(-x, -y)

idcsv <- left_join(idcsv, first, by = "first_sample_id")
idcsv <- left_join(idcsv, sec, by = "second_sample_id")

# mark all of the rows where the genetic recapture is cofirmed by  size, location
idcsv <- idcsv %>% 
  mutate(mark = ifelse(first_gen_id == second_gen_id, "apparent", "questionable"), 
         mark = ifelse(is.na(mark), "questionable", mark))

dist <- idcsv %>% 
  select(first_id, second_id, distkm, mark) %>% 
  group_by(distkm, mark) %>% 
  summarise(count = n())

# convert to meters
dist <- dist %>%
  mutate(distm = distkm * 1000)

```

Ignore the questionable far distance fish
```{r}
dist_sm <- dist %>% 
  filter(distm < 1000)

ggplot(dist_sm, aes(x = mark, y = distm)) + 
  geom_boxplot() +
  coord_flip() +
  geom_jitter()
```

Why are there still 2 dots below 100m for questionable? Because they are NA's because of the regenotype concatenation in the naming scheme.
```{r}
x <- dist_sm %>% 
  filter(mark == "questionable", distkm < .1)

idcsv %>% 
  filter(distkm == x$distkm[1] | distkm == x$distkm[2])

z <- tibble(sample_id = c("APCL17_080", "APCL16_037", "APCL15_374004"))

# do these fish have any other matches?
temp <- fish %>% 
  filter(sample_id %in% z$sample_id) %>% 
  select(sample_id, gen_id)

# no other matches
temp1 <- fish %>% 
  filter(gen_id == temp$gen_id[3])

temp <- temp %>% 
  mutate(gen_id = 2917)

# pull in the clownfish table from the database
fish <- leyte %>% 
  tbl("clownfish") %>% 
  collect()

# The following line didn't work, some sample_ids were not changed!!!
# fish <- fish %>%
# mutate(gen_id = ifelse(sample_id %in% fix_list$sample_id, fix_list$gen_id, gen_id))

fish <- fish %>% 
  mutate(gen_id = ifelse(sample_id == "APCL17_080", 2917, gen_id))


 # backup db
  ley <- write_db("Leyte")
  clown <- ley %>% 
    tbl("clownfish") %>% 
    collect()
  #write_csv(clown, path = paste0("../db_backups/", Sys.time(), "_clownfish_db.csv"))
  
    # write change
# #dbWriteTable(ley, "clownfish", fish, row.names = F, overwrite = T)
# dbDisconnect(ley)
```
Try graphs again
# Generate plots of distribution of distance between captures for obvious recaptures and flagged recaptures
```{r plots 4}
idcsv <- read_csv("data/seq03-33_identity/33-03_seq_identity_big_table.csv")

# mark rows where the match was considered an obvious recapture based on genetics, size, lat lon.

# get the most current gen_ids for the first samples 
first <- fish_anem_dive(idcsv$first_sample_id, leyte)
 names(first) <- paste0("first_", names(first)) 
 
 # remove the sample_id from this group of names to use later
 f_except <- select(first, -first_sample_id)
 x <- names(f_except)

# get the most current gen_ids for the second samples
sec <- fish_anem_dive(idcsv$second_sample_id, leyte)
names(sec) <- paste0("second_", names(sec))

# remove the sample_id from this group of names to use later
s_except <- select(sec, -second_sample_id)
y <- names(s_except)

# remove the above columns from idcsv so that the most updated version can be reattached
idcsv <- idcsv %>% 
  select(-x, -y)

idcsv <- left_join(idcsv, first, by = "first_sample_id")
idcsv <- left_join(idcsv, sec, by = "second_sample_id")

# mark all of the rows where the genetic recapture is cofirmed by  size, location
idcsv <- idcsv %>% 
  mutate(mark = ifelse(first_gen_id == second_gen_id, "apparent", "questionable"), 
         mark = ifelse(is.na(mark), "questionable", mark))

dist <- idcsv %>% 
  select(first_id, second_id, distkm, mark) %>% 
  group_by(distkm, mark) %>% 
  summarise(count = n())

# convert to meters
dist <- dist %>%
  mutate(distm = distkm * 1000)

```

Ignore the questionable far distance fish
```{r}
dist_sm <- dist %>% 
  filter(distm < 1000)

ggplot(dist_sm, aes(x = mark, y = distm)) + 
  geom_boxplot() +
  coord_flip() +
  geom_jitter()
```
```{r}
ggplot(dist, aes(distm, fill = mark)) +
  geom_histogram(binwidth = 500, position = "dodge") +
  scale_y_continuous(limits = c(0, 5))+
  theme_bw()+
  theme(
    plot.background = element_blank()
    ,panel.grid.major = element_blank()
    ,panel.grid.minor = element_blank()
    ,panel.border = element_blank()
  ) 
```






