---
title: "Known issues investigation"
output: html_notebook
---
This notebook writes reports on why samples are in the known issues table and double checks their status after a sequencing run.
```{r setup}
source("scripts/gen_helpers.R")
leyte <- read_db("Leyte")
lab <- read_db("Laboratory")
```



# Pull in the known issues list
```{r known_issues}
ki <- lab %>% 
  tbl("known_issues") %>% 
  collect() %>% 
  distinct()
```

# are there any duplicate ligation ids?
```{r dups}
dups <- ki %>% 
  group_by(ligation_id) %>% 
  summarise(count = n()) %>% 
  filter(count > 1) 

# compare the issues for the dups to see why they are different
dups <- left_join(dups, ki, by = "ligation_id") %>% 
  distinct()

```
Couldn't look at the full notes in the issues so examines the dups in sequel pro and deleted the obsolete rows.
Writing cleaned ki to db
```{r write_to_db}
rm(dups)
lb <- write_db("Laboratory")
dbWriteTable(lb, "known_issues", ki, overwrite=T, row.names=F)
dbDisconnect(lb)
rm(lb)
```

# Sample by sample examination
```{r 1}
i <- 1
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")


```

L0288 was red-flagged because matches L0606 with a distance error (~600m), lat longs check out, regenotype, does not appear in ID results for SEQ17. However, it doesn’t appear in identity analysis results for seq17 or seq33. Making a note in Evernote but removing from known issues list. **Update** L0288 was not in the seq33 id results because it was moved to problem samples.  Moving back into the general population on amphiprion.


```{r 2}
i <- 2
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])
```
L0908 was red-flagged because This is from a contaminated digest, do not use. This sample was contaiminated. Logging into amphiprion and moving the sample out of the pool of potential samples for analysis. L0909 is the same. L3115 is the same.

```{r 5}
i <- 5
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])
```
L3008 was red-flagged because Only ~3000 reads, failed seq. This sample doesn’t need to be on this list because it won’t pass through a filtering run into a genepop. Making sure it is in the general population of samples on amphiprion.

```{r 6}
i <- 6
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0370 was red-flagged because matches L0742 with a distance error (~1.8km), lat longs check out, has not been regenotyped.  Can remove from the known issues list because it doesn't appear on seq 33 ID analysis.  Making note in evernote. **Update** L0370 was not in the seq33 id results because it was moved to problem samples.  Moving back into the general population on amphiprion.

```{r 7}
i <- 7
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L2988 was red-flagged because distance error, potential label issue.  This sample has been regenotyped but wasn't in the general population of samples for SNP calling.

```{r 8}
i <- 8
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0377 was red-flagged because Distance error, matches with L0974, regenotype. Moved back into the general population of samples.

```{r 9}
i <- 9
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0413 was red-flagged because This is not APCL13_040. It is a fish caught in 2013 at Elementary School, probably one of APCL13_105-121 (those were on the sample plate). Can’t be 119 or 120 because those were larger than 11cm and this fish was 9.6 and 10.5 cm when caught in 2015,2016.  This sample was in the problem samples folder, moved back into the general population of samples.
```{r 10}
i <- 10
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0415 was red-flagged because distance error, matches with L2935, compare fish on anemones to see if there is a labeling error. Moved the concatenated regenotype sample back into the general population.

```{r 11}
i <- 11
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0425 was red-flagged because lab error, matches exactly to L0427, until one of these matches to a different fish cannot know which one it really is.

```{r 12}
i <- 12
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0427 was red-flagged because lab error, matches exactly to L0425, until one of these matches to a different fish cannot know which one it really is.

```{r 13}
i <- 13
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0489 was red-flagged because extract plate label error, could be APCL13_535 (consecutive) or could be APCL13_537 if APCL13_534,APCL13_535 were mistype to 544, 545.

```{r 14}
i <- 14
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0490 was red-flagged because extract plate label error, could be APCL13_537 (consecutive) or could be APCL13_539 if APCL13_534,APCL13_535 were mistype to 544, 545.

```{r 15}
i <- 15
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0491 was red-flagged because extract plate label error, could be APCL13_538 (consecutive) or could be APCL13_540 if APCL13_534,APCL13_535 were mistype to 544, 545.

```{r 16}
i <- 16
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0493 was red-flagged because extract plate label error, could be APCL13_549 (consecutive) or could be APCL13_547 (strange tube order in freezer box).

```{r 17}
i <- 17
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L1607 was red-flagged because There was a label shift on this plate and this ligation does not match to the previous genotype - L1042, this does not appear to be APCL13_059, L1607 shows up as a potential parent for 18 offspring

```{r 18}
i <- 18
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0595 was red-flagged because extract plate label error, could be 304 or 306.

```{r 19}
i <- 19
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")

test <- full_meta(c("APCL14_305", "APCL14_306"), leyte)
```
L0596 was red-flagged because Unsure which fin clip was placed in this extraction, APCL14_305 or APCL14_306, matches to APCL18_327 from Sitio Baybayon.  Because both fish were caught at the same time in the same place, can't tell which one it is.

```{r 20}
i <- 20
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L0610 was red-flagged because distance error, matches with L2900, possible extract label issue.

```{r 21}
i <- 21
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L3145 was red-flagged because Distance error, potiential extract labeling issue.  The original genotype L0936 matches to 3 other captures from Palanas.  Need to re-run the snp analysis with L3145 in it to see if there is still and issue with this sample.

```{r 22}
i <- 22
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])

# has this sample been ligated more than once?
  # get sample id
y <- samp_from_lig(x)
  # get work history for this sample
z <- work_history(y, "sample_id")
```
L3122 was red-flagged because distance error, matches with L2754, potential extract label issue.

```{r 23}
for (i in seq(ki$ligation_id)){
x <- ki %>% 
  filter(ligation_id == ki$ligation_id[i])
cat(i, " ", x$ligation_id[1], "was red-flagged because ", x$issue,
    "\n")

}
```


