---
title: "Determine Cervus Parameters"
date: 2019-05-09
output:
  html_notebook:
    code_folding: hide
    df_print: paged
    highlight: kate
---
#### Cervus requires hard numbers instead of percentages for cutoffs in identity analysis.  We enter 80% of the total loci in the analysis for matching loci and 10% mismatch. However, some fish have missing data and so 80%/10% of *their* total loci is much less than the numbers that are hard coded into the analysis and potential recaptures are ignored.  


```{r setup, include=FALSE, message=FALSE}
knitr::opts_chunk$set(
	eval = FALSE,
	message = FALSE,
	warning = FALSE,
	include = FALSE
)
# load libraries
library(lubridate)
library(tidyverse)
library(here)
library(clownfish)
library(geosphere)
library(janitor)

# while db connection using helper file isn't working 
source("~/Documents/clownfish-pkg/R/db_connections.R")
leyte <- read_db("Leyte")
lab <- read_db("Laboratory")
```

Our dataset is made up of 2774 individuals and a total of 1447 loci.  We allowed individuals with 30% missing data during SNP filtering.

# Test 1 
- 80% matching loci (1157), 10% mismatching allowed (145) resulted in 403 proposed recapture events.

This test potentially misses some recaptures because individuals are missing more data than the 20% cutoff (80% data must be present to be considered for a recapture event).


# Test 2 
- 70% matching loci (1013), 10% mismatching allowed (145) resulted in 422 proposed recapture events.  This increase in individuals makes sense because we are now including individuals that had more missing data.


# Test 3 
- 60% matching loci (868), 10% mismatching allowed (145) resulted in 427 proposed recapture events. This slight jump might represent individuals who were on the border of the 30% missing data cutoff (for example they might've had 69% matching loci).


# Test 4 
- 50% matching loci (724), 10% mismatching allowed (145) resulted in 428 proposed recapture events. Not a big increase because we got most of the samples that are missing data at the 70% mark (where the missing data cutoff was for filtering)

# Test 5
- 10% matching loci (146), 10% mismatching allowed (145) resulted in 428 proposed recapture events. No change because the 50% matching loci captured all of the events.

# Test 6
- 50% matching loci (724), 10% mismatching of the lowest number of loci allowed (108, equivalent to 7% of the total number of loci, still higher than genotyping error) resulted in 423 proposed recapture events. Fewer recapture events makes sense because the mismatch is more conservative.

# Test 7
- 50% matching loci (724), 5% mismatching allowed (72), which is lower than the genotype error rate resulted in  proposed recapture events. This resulted in 377 recapture events.  Because this number is lower than the genotype error, it is expected that this is missing some true recaptures.

```{r}
idcsv_raw <- read_csv("https://raw.githubusercontent.com/pinskylab/genomics/master/data/seq03-33_identity/33-03_seq_identity_ID.csv", col_types = cols(
  `First ID` = col_character(),
  `Loci typed` = col_integer(),
  `Second ID` = col_character(),
  `Loci typed_1` = col_integer(),
  `Matching loci` = col_integer(),
  `Mismatching loci` = col_integer(),
  pID = col_double(),
  pIDsib = col_double(),
  Status = col_character()
)) 

names(idcsv_raw) <- c("first_id",  "first_loci_typed",  "second_id",  "second_loci_typed",  "matching_loci",  "mismatching_loci",  "p_id",  "p_id_sib",  "status")

# add mismatch rate
idcsv <- idcsv_raw %>%
  mutate(loci_compared = mismatching_loci + matching_loci,
    mismatch_prop = mismatching_loci/loci_compared) %>% 
  # order by number of matching loci
  arrange(matching_loci) 

ggplot(idcsv, aes(x = loci_compared, y = mismatch_prop)) +
  geom_point() +
  theme_bw()
```
